\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
  \newcommand{\euro}{â‚¬}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}


\usepackage{longtable,booktabs}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

\RequirePackage[]{C:/Users/jrgonzalez/Documents/R/win-library/3.6/BiocStyle/resources/tex/Bioconductor}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

\bioctitle[]{Chronological and gestational DNAm age estimation using different methylation-based clocks}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
  \subtitle{Gerardo Alfonso, Dolors Pelegri and Juan R Gonzalez}
\author{Institute for Global Health (ISGlobal), Barcelona, Spain
Bioinformatics Research Group in Epidemiolgy (BRGE)
http://brge.isglobal.org
}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{2020-01-15}


% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

% code highlighting
\definecolor{fgcolor}{rgb}{0.251, 0.251, 0.251}
\newcommand{\hlnum}[1]{\textcolor[rgb]{0.816,0.125,0.439}{#1}}%
\newcommand{\hlstr}[1]{\textcolor[rgb]{0.251,0.627,0.251}{#1}}%
\newcommand{\hlcom}[1]{\textcolor[rgb]{0.502,0.502,0.502}{\textit{#1}}}%
\newcommand{\hlopt}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlstd}[1]{\textcolor[rgb]{0.251,0.251,0.251}{#1}}%
\newcommand{\hlkwa}[1]{\textcolor[rgb]{0.125,0.125,0.941}{#1}}%
\newcommand{\hlkwb}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlkwc}[1]{\textcolor[rgb]{0.251,0.251,0.251}{#1}}%
\newcommand{\hlkwd}[1]{\textcolor[rgb]{0.878,0.439,0.125}{#1}}%
\let\hlipl\hlkwb
%
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
%
\newenvironment{Shaded}{\begin{myshaded}}{\end{myshaded}}
% set background for result chunks
\let\oldverbatim\verbatim
\renewenvironment{verbatim}{\color{codecolor}\begin{myshaded}\begin{oldverbatim}}{\end{oldverbatim}\end{myshaded}}
%
\newcommand{\KeywordTok}[1]{\hlkwd{#1}}
\newcommand{\DataTypeTok}[1]{\hlkwc{#1}}
\newcommand{\DecValTok}[1]{\hlnum{#1}}
\newcommand{\BaseNTok}[1]{\hlnum{#1}}
\newcommand{\FloatTok}[1]{\hlnum{#1}}
\newcommand{\ConstantTok}[1]{\hlnum{#1}}
\newcommand{\CharTok}[1]{\hlstr{#1}}
\newcommand{\SpecialCharTok}[1]{\hlstr{#1}}
\newcommand{\StringTok}[1]{\hlstr{#1}}
\newcommand{\VerbatimStringTok}[1]{\hlstr{#1}}
\newcommand{\SpecialStringTok}[1]{\hlstr{#1}}
\newcommand{\ImportTok}[1]{{#1}}
\newcommand{\CommentTok}[1]{\hlcom{#1}}
\newcommand{\DocumentationTok}[1]{\hlcom{#1}}
\newcommand{\AnnotationTok}[1]{\hlcom{#1}}
\newcommand{\CommentVarTok}[1]{\hlcom{#1}}
\newcommand{\OtherTok}[1]{{#1}}
\newcommand{\FunctionTok}[1]{\hlstd{#1}}
\newcommand{\VariableTok}[1]{\hlstd{#1}}
\newcommand{\ControlFlowTok}[1]{\hlkwd{#1}}
\newcommand{\OperatorTok}[1]{\hlopt{#1}}
\newcommand{\BuiltInTok}[1]{{#1}}
\newcommand{\ExtensionTok}[1]{{#1}}
\newcommand{\PreprocessorTok}[1]{\textit{#1}}
\newcommand{\AttributeTok}[1]{{#1}}
\newcommand{\RegionMarkerTok}[1]{{#1}}
\newcommand{\InformationTok}[1]{\textcolor{messagecolor}{#1}}
\newcommand{\WarningTok}[1]{\textcolor{warningcolor}{#1}}
\newcommand{\AlertTok}[1]{\textcolor{errorcolor}{#1}}
\newcommand{\ErrorTok}[1]{\textcolor{errorcolor}{#1}}
\newcommand{\NormalTok}[1]{\hlstd{#1}}
%
\AtBeginDocument{\bibliographystyle{C:/Users/jrgonzalez/Documents/R/win-library/3.6/BiocStyle/resources/tex/unsrturl}}

\begin{document}
\maketitle

\packageVersion{methylclock 0.5.0}

{
\setcounter{tocdepth}{2}
\tableofcontents
\newpage
}
\hypertarget{description-of-implemented-clocks}{%
\section{Description of implemented clocks}\label{description-of-implemented-clocks}}

This manual describes how to estimate chronological and gestational DNA methylation (DNAm) age as well as biological age using different methylation clocks. The package includes the following estimators:

\hypertarget{chronological-dnam-age-in-years}{%
\subsection{Chronological DNAm age (in years)}\label{chronological-dnam-age-in-years}}

\begin{itemize}
\tightlist
\item
  \textbf{Horvath's clock}: It uses 353 CpGs described in Horvath (2013). It was trained using 27K and 450K arrays in samples from different tissues. Other three different age-related biomarkers are also computed:

  \begin{itemize}
  \tightlist
  \item
    \textbf{AgeAcDiff} (DNAmAge acceleration difference): Difference between DNAmAge and chronological age.
  \item
    \textbf{IEAA} (Intrinsic Epigenetic Age Acceleration): Residuals obtained after regressing DNAmAge and chronological age adjusted by cell counts.\\
  \item
    \textbf{EEAA} (Extrinsic Epigenetic Age Acceleration): Residuals obtained after regressing DNAmAge and chronological age. This measure was also known as DNAmAge acceleration residual in the first Horvath's paper.
  \end{itemize}
\item
  \textbf{Hannum's clock}: It uses 71 CpGs described in Hannum et al. (2013). It was trained using 450K array in blood samples. Another are-related biomarer is also computed:

  \begin{itemize}
  \tightlist
  \item
    \textbf{AMAR} (Apparent Methylomic Aging Rate): Measure proposed in Hannum et al. (2013) computed as the ratio between DNAm age and the chronological age.
  \end{itemize}
\item
  \textbf{BNN}: It uses Horvath's CpGs to train a Bayesian Neural Network (BNN) to predict DNAm age as described in Alfonso and Gonzalez (2018).
\item
  \textbf{Horvath's skin+blood clock (Horvath2)}: Epigenetic clock for skin and blood cells. It uses 391 CpGs described in Horvath et al. (2018). It was trained using 450K EPIC arrays in skin and blood sampels.
\item
  \textbf{PedBE clock}: Epigenetic clock from buccal epithelial swabs. It's intended purpose is buccal samples from individuals aged 0-20 years old. It uses 84 CpGs described in McEwen et al. (2019). The authors gathered 1,721 genome-wide DNAm profiles from 11 different cohorts with individuals aged 0 to 20 years old.
\end{itemize}

\hypertarget{gestational-dnam-age-in-weeks}{%
\subsection{Gestational DNAm age (in weeks)}\label{gestational-dnam-age-in-weeks}}

\begin{itemize}
\tightlist
\item
  \textbf{Knight's clock}: It uses 148 CpGs described in Knight et al. (2016). It was trained using 27K and 450K arrays in coord blood samples.
\item
  \textbf{Bohlin's clock}: It uses 96 CpGs described in Bohlin et al. (2016). It was trained using 450K array in coord blood samples.
\item
  \textbf{Mayne's clock}: It uses 62 CpGs described in Mayne et al. (2017). It was trained using 27K and 450K.
\item
  \textbf{Lee's clocks}: Three different biological clocks described in Lee et al. (2019) are implemented. It was trained for 450K and EPIC arrays in placenta samples.

  \begin{itemize}
  \tightlist
  \item
    \textbf{RPC clock}: Robust placental clock (RPC). It uses 558 CpG sites.
  \item
    \textbf{CPC clock}: Control placental clock (CPC). It usses 546 CpG sites.
  \item
    \textbf{Refined RPC clock}: Useful for uncomplicated term pregnancies (e.g.~gestational age \textgreater{}36 weeks). It uses 396 CpG sites.
  \end{itemize}
\end{itemize}

The biological DNAm clocks implemented in our package are:

\begin{itemize}
\tightlist
\item
  \textbf{Levine's clock} (also know as PhenoAge): It uses 513 CpGs described in Levine et al. (2018). It was trained using 27K, 450K and EPIC arrays in blood samples.
\end{itemize}

The main aim of this package is to facilitate the interconnection with R and Bioconductor's infrastructure and, hence, avoiding submitting data to online calculators. Additionally, \texttt{methylclock} also provides an unified way of computing DNAm age to help downstream analyses.

\hypertarget{getting-started}{%
\section{Getting started}\label{getting-started}}

The package depends on some R packages that can be previously installed into your computer by:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(BiocManager)}
\KeywordTok{install}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\StringTok{"tidyverse"}\NormalTok{, }\StringTok{"impute"}\NormalTok{, }\StringTok{"Rcpp"}\NormalTok{, }\StringTok{"GAprediction"}\NormalTok{))}

\KeywordTok{library}\NormalTok{(devtools)}
\KeywordTok{install_github}\NormalTok{(}\StringTok{"perishky/meffil"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Then \texttt{methylclock} package is installed into your computer by executing:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{install_github}\NormalTok{(}\StringTok{"isglobal-brge/methylclock"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

The package is loaded into R as usual:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(methylclock)}
\end{Highlighting}
\end{Shaded}

These libraries are required to reproduce this document:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(Biobase)}
\KeywordTok{library}\NormalTok{(tibble)}
\KeywordTok{library}\NormalTok{(ggplot2)}
\KeywordTok{library}\NormalTok{(ggpmisc)}
\KeywordTok{library}\NormalTok{(GEOquery)}
\end{Highlighting}
\end{Shaded}

\hypertarget{dna-methylation-clocks}{%
\section{DNA Methylation clocks}\label{dna-methylation-clocks}}

The main function to estimate chronological and biological mDNA age is called \texttt{DNAmAge} while the gestational DNAm age is estimated using \texttt{DNAmGA} function. Both functions have similar input arguments. Next subsections detail some of the important issues to be consider before computind DNAm clocks.

\hypertarget{data-format}{%
\subsection{Data format}\label{data-format}}

The methylation data is given in the argument \texttt{x}. They can be either beta or M values. The argument \texttt{toBetas} should be set to TRUE when M values are provided. The \texttt{x} object can be:

\begin{itemize}
\item
  A \textbf{matrix} with CpGs in rows and individuals in columns having the name of the CpGs in the rownames.
\item
  A \textbf{data frame} or a \textbf{tibble} with CpGs in rows and individuals in columns having the name of the CpGs in the first column (e.g.~cg00000292, cg00002426, cg00003994, \ldots{}) as required in the Horvath's DNA Methylation Age Calculator website (\url{https://dnamage.genetics.ucla.edu/home}).
\item
  A \textbf{GenomicRatioSet} object, the default method to encapsulate methylation data in \texttt{minfi} Bioconductor package.
\item
  An \textbf{ExpressionSet} object as obtained, for instance, when downloading methylation data from GEO (\url{https://www.ncbi.nlm.nih.gov/geo/}).
\end{itemize}

\hypertarget{data-nomalization}{%
\subsection{Data nomalization}\label{data-nomalization}}

In principle, data can be normalized by using any of the existing standard methods such as QN, ASMN, PBC, SWAN, SQN, BMIQ (see a revision of those methods in Wang et al. (2015)). \texttt{DNAmAge} function includes the BMIQ method proposed by Teschendorff et al. (2012) using Horvath's robust implementation that basically consists of an optimal R code implementation and optimization procedures. This normalization is recommended by Horvath since it improves the predictions for his clock. This normalization procedure is very time-consuming. In order to overcome these difficulties, we have parallelize this process using \texttt{BiocParallel} library. This step is not mandatory, so that, you can use your normalized data and set the argument \texttt{normalize} equal to FALSE (default).

\hypertarget{missing-individuals-data}{%
\subsection{Missing individual's data}\label{missing-individuals-data}}

All the implemented methods require complete cases. \texttt{DNAmAge} function has an imputation method based on KNN implemented in the function \texttt{knn.impute} from \texttt{impute} Bioconductor package. This is performed when missing data is present in the CpGs used in any of the computed clocks. There is also another option based on a fast imputation method that imputes missing values by the median of required CpGs as recommended in Bohlin et al. (2016). This is recommended when analyzing 450K arrays since \texttt{knn.impute} for large datasets may be very time consuming. Fast imputation can be performed by setting \texttt{fastImp=TRUE} which is not the default value.

\hypertarget{missing-cpgs-of-dnam-clocks}{%
\subsection{Missing CpGs of DNAm clocks}\label{missing-cpgs-of-dnam-clocks}}

By default the package computes the different clocks when there are more than 80\% of the required CpGs of each method. Nothing is required when having missing CpGs since the main functions will return NA for those estimators when this criteria is not meet. Let us use a test dataset (\texttt{TestDataset}) which is available within the package to illustrate the type of information we are obtaining:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{cpgs.missing <-}\StringTok{ }\KeywordTok{checkClocks}\NormalTok{(TestDataset)}
\NormalTok{          clock Cpgs_in_clock missing_CpGs percentage}
  \DecValTok{1}\NormalTok{     Horvath           }\DecValTok{354}            \DecValTok{2}        \FloatTok{0.6}
  \DecValTok{2}\NormalTok{      Hannum            }\DecValTok{71}           \DecValTok{64}       \FloatTok{90.1}
  \DecValTok{3}\NormalTok{      Levine           }\DecValTok{514}            \DecValTok{3}        \FloatTok{0.6}
  \DecValTok{4}\NormalTok{ SkinHorvath           }\DecValTok{392}          \DecValTok{283}       \FloatTok{72.2}
  \DecValTok{5}\NormalTok{       PedBE            }\DecValTok{95}           \DecValTok{91}       \FloatTok{95.8}
\NormalTok{  There are some clocks that cannot be computed since your data do not contain the required CpGs. }
\NormalTok{          These are the total number of missing CpGs }\ControlFlowTok{for}\NormalTok{ each clock }\OperatorTok{:}\StringTok{ }
\StringTok{  }
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{cpgs.missing.GA <-}\StringTok{ }\KeywordTok{checkClocksGA}\NormalTok{(TestDataset)}
\NormalTok{     clock Cpgs_in_clock missing_CpGs percentage}
  \DecValTok{1}\NormalTok{ Knight           }\DecValTok{149}            \DecValTok{0}        \FloatTok{0.0}
  \DecValTok{2}\NormalTok{ Bohlin            }\DecValTok{96}           \DecValTok{87}       \FloatTok{90.6}
  \DecValTok{3}\NormalTok{  Mayne            }\DecValTok{63}            \DecValTok{0}        \FloatTok{0.0}
  \DecValTok{4}\NormalTok{    Lee          }\DecValTok{1126}         \DecValTok{1072}       \FloatTok{95.2}
\NormalTok{  There are some clocks that cannot be computed since your data do not contain the required CpGs. }
\NormalTok{          These are the total number of missing CpGs }\ControlFlowTok{for}\NormalTok{ each clock }\OperatorTok{:}\StringTok{ }
\StringTok{   }
\StringTok{     }\NormalTok{clock Cpgs_in_clock missing_CpGs percentage}
  \DecValTok{1}\NormalTok{ Knight           }\DecValTok{149}            \DecValTok{0}        \FloatTok{0.0}
  \DecValTok{2}\NormalTok{ Bohlin            }\DecValTok{96}           \DecValTok{87}       \FloatTok{90.6}
  \DecValTok{3}\NormalTok{  Mayne            }\DecValTok{63}            \DecValTok{0}        \FloatTok{0.0}
  \DecValTok{4}\NormalTok{    Lee          }\DecValTok{1126}         \DecValTok{1072}       \FloatTok{95.2}
\end{Highlighting}
\end{Shaded}

The objects \texttt{cpgs.missing} and \texttt{cpgs.missing.GA} are lists havint the missing CpGs of each clock

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{names}\NormalTok{(cpgs.missing)}
\NormalTok{  [}\DecValTok{1}\NormalTok{] }\StringTok{"Horvath"}  \StringTok{"Hannum"}   \StringTok{"Levine"}   \StringTok{"Horvath2"} \StringTok{"PedBE"}
\NormalTok{cpgs.missing}\OperatorTok{$}\NormalTok{Hannum}
\NormalTok{   [}\DecValTok{1}\NormalTok{] }\StringTok{"cg20822990"}      \StringTok{"cg22512670"}      \StringTok{"cg25410668"}      \StringTok{"cg04400972"}     
\NormalTok{   [}\DecValTok{5}\NormalTok{] }\StringTok{"cg16054275"}      \StringTok{"cg10501210"}      \StringTok{"ch.2.30415474F"}  \StringTok{"cg22158769"}     
\NormalTok{   [}\DecValTok{9}\NormalTok{] }\StringTok{"cg02085953"}      \StringTok{"cg06639320"}      \StringTok{"cg22454769"}      \StringTok{"cg24079702"}     
\NormalTok{  [}\DecValTok{13}\NormalTok{] }\StringTok{"cg23606718"}      \StringTok{"cg22016779"}      \StringTok{"cg03607117"}      \StringTok{"cg07553761"}     
\NormalTok{  [}\DecValTok{17}\NormalTok{] }\StringTok{"cg00481951"}      \StringTok{"cg25478614"}      \StringTok{"cg25428494"}      \StringTok{"cg02650266"}     
\NormalTok{  [}\DecValTok{21}\NormalTok{] }\StringTok{"cg08234504"}      \StringTok{"cg23500537"}      \StringTok{"cg20052760"}      \StringTok{"cg16867657"}     
\NormalTok{  [}\DecValTok{25}\NormalTok{] }\StringTok{"cg06685111"}      \StringTok{"cg00486113"}      \StringTok{"cg13001142"}      \StringTok{"cg20426994"}     
\NormalTok{  [}\DecValTok{29}\NormalTok{] }\StringTok{"cg14361627"}      \StringTok{"cg08097417"}      \StringTok{"cg07955995"}      \StringTok{"cg22285878"}     
\NormalTok{  [}\DecValTok{33}\NormalTok{] }\StringTok{"cg03473532"}      \StringTok{"cg08540945"}      \StringTok{"cg07927379"}      \StringTok{"cg16419235"}     
\NormalTok{  [}\DecValTok{37}\NormalTok{] }\StringTok{"cg07583137"}      \StringTok{"cg22796704"}      \StringTok{"cg19935065"}      \StringTok{"cg23091758"}     
\NormalTok{  [}\DecValTok{41}\NormalTok{] }\StringTok{"cg23744638"}      \StringTok{"cg04940570"}      \StringTok{"cg11067179"}      \StringTok{"cg22213242"}     
\NormalTok{  [}\DecValTok{45}\NormalTok{] }\StringTok{"cg06419846"}      \StringTok{"cg02046143"}      \StringTok{"cg00748589"}      \StringTok{"cg18473521"}     
\NormalTok{  [}\DecValTok{49}\NormalTok{] }\StringTok{"cg01528542"}      \StringTok{"ch.13.39564907R"} \StringTok{"cg03032497"}      \StringTok{"cg04875128"}     
\NormalTok{  [}\DecValTok{53}\NormalTok{] }\StringTok{"cg09651136"}      \StringTok{"cg03399905"}      \StringTok{"cg04416734"}      \StringTok{"cg07082267"}     
\NormalTok{  [}\DecValTok{57}\NormalTok{] }\StringTok{"cg14692377"}      \StringTok{"cg06874016"}      \StringTok{"cg21139312"}      \StringTok{"cg02867102"}     
\NormalTok{  [}\DecValTok{61}\NormalTok{] }\StringTok{"cg19283806"}      \StringTok{"cg14556683"}      \StringTok{"cg07547549"}      \StringTok{"cg08415592"}
\end{Highlighting}
\end{Shaded}

\hypertarget{cell-counts}{%
\subsection{Cell counts}\label{cell-counts}}

The EEAA method requires to estimate cell counts. We use the package \texttt{meffil} (Min et al. (2018)) that provides some functions to estimate cell counts using predefined datasets. This is performed by setting \texttt{cell.count=TRUE} (default value). The reference panel is passed through the argument \texttt{cell.count.reference}. So far, the following options are available:

\begin{itemize}
\tightlist
\item
  \textbf{``blood gse35069 complete''}: methylation profiles from Reinius et al. (2012) for purified blood cell types. It includes CD4T, CD8T, Mono, Bcell, NK, Neu and Eos.
\item
  \textbf{``blood gse35069''}: methylation profiles from Reinius et al. (2012) for purified blood cell types. It includes CD4T, CD8T, Mono, Bcell, NK and Gran.
\item
  \textbf{``blood gse35069 chen''}: methylation profiles from Chen et al. (2017) blood cell types. It includes CD4T, CD8T, Mono, Bcell, NK, Neu and Eos.
\item
  \textbf{``andrews and bakulski cord blood''}. Cord blood reference from Bakulski et al. (2016). It includes Bcell, CD4T, CD8T, Gran, Mono, NK and nRBC.
\item
  \textbf{``cord blood gse68456''} Cord blood methylation profiles from Goede et al. (2015). It includes CD4T, CD8T, Mono, Bcell, NK, Neu, Eos and RBC.
\item
  \textbf{``gervin and lyle cord blood''} Cord blood reference generated by Kristina Gervin and Robert Lyle, available at \texttt{miffil} package. It includes CD14, Bcell, CD4T, CD8T, NK, Gran.
\item
  \textbf{``saliva gse48472''}: Reference generated from the multi-tissue pannel from Slieker et al. (2013). It includes Buccal, CD4T, CD8T, Mono, Bcell, NK, Gran.
\end{itemize}

\hypertarget{chronological-and-biological-dnam-age-estimation}{%
\section{Chronological and biological DNAm age estimation}\label{chronological-and-biological-dnam-age-estimation}}

Next we illustrate how to estimate the chronological DNAm age using several datasets which aim to cover different data input formats.

\hypertarget{data-in-horvaths-format-e.g.-csv-with-cpgs-in-rows}{%
\subsection{\texorpdfstring{Data in Horvath's format (e.g.~\texttt{csv} with CpGs in rows)}{Data in Horvath's format (e.g.~csv with CpGs in rows)}}\label{data-in-horvaths-format-e.g.-csv-with-cpgs-in-rows}}

Let us start by reproducing the results proposed in Horvath (2013). It uses the format available in the file 'MethylationDataExample55.csv" from his tutorial (available \href{https://dnamage.genetics.ucla.edu/home}{here}). These data are available at \texttt{methylclock} package. Although these data can be loaded into R by using standard functions such as \texttt{read.csv} we hihgly recommend to use functions from \texttt{tidiverse}, in particular \texttt{read\_csv} from \texttt{readr} package. The main reason is that currently researchers are analyzing Illumina 450K or EPIC arrays that contains a huge number of CpGs that can take a long time to be loaded when using basic importing R function. These functions import \texttt{csv} data as tibble which is one of the possible formats of \texttt{DNAmAge} function

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(tidyverse)}
\NormalTok{path <-}\StringTok{ }\KeywordTok{system.file}\NormalTok{(}\StringTok{"extdata"}\NormalTok{, }\DataTypeTok{package =} \StringTok{"methylclock"}\NormalTok{)}
\NormalTok{MethylationData <-}\StringTok{ }\KeywordTok{read_csv}\NormalTok{(}\KeywordTok{file.path}\NormalTok{(path, }\StringTok{"MethylationDataExample55.csv"}\NormalTok{))}
\NormalTok{MethylationData}
  \CommentTok{# A tibble: 27,578 x 17}
\NormalTok{     ProbeID GSM946048 GSM946049 GSM946052 GSM946054 GSM946055 GSM946056}
     \OperatorTok{<}\NormalTok{chr}\OperatorTok{>}\StringTok{       }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{     }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{     }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{     }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{     }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{     }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}
\StringTok{   }\DecValTok{1}\NormalTok{ cg0000}\OperatorTok{~}\StringTok{    }\FloatTok{0.706}    \FloatTok{0.730}     \FloatTok{0.705}     \FloatTok{0.751}     \FloatTok{0.715}     \FloatTok{0.634}  
   \DecValTok{2}\NormalTok{ cg0000}\OperatorTok{~}\StringTok{    }\FloatTok{0.272}    \FloatTok{0.274}     \FloatTok{0.311}     \FloatTok{0.279}     \FloatTok{0.178}     \FloatTok{0.269}  
   \DecValTok{3}\NormalTok{ cg0000}\OperatorTok{~}\StringTok{    }\FloatTok{0.0370}   \FloatTok{0.0147}    \FloatTok{0.0171}    \FloatTok{0.0290}    \FloatTok{0.0163}    \FloatTok{0.0243} 
   \DecValTok{4}\NormalTok{ cg0000}\OperatorTok{~}\StringTok{    }\FloatTok{0.133}    \FloatTok{0.120}     \FloatTok{0.121}     \FloatTok{0.107}     \FloatTok{0.110}     \FloatTok{0.129}  
   \DecValTok{5}\NormalTok{ cg0000}\OperatorTok{~}\StringTok{    }\FloatTok{0.0309}   \FloatTok{0.0192}    \FloatTok{0.0217}    \FloatTok{0.0132}    \FloatTok{0.0181}    \FloatTok{0.0243} 
   \DecValTok{6}\NormalTok{ cg0000}\OperatorTok{~}\StringTok{    }\FloatTok{0.0700}   \FloatTok{0.0715}    \FloatTok{0.0655}    \FloatTok{0.0719}    \FloatTok{0.0914}    \FloatTok{0.0508} 
   \DecValTok{7}\NormalTok{ cg0000}\OperatorTok{~}\StringTok{    }\FloatTok{0.993}    \FloatTok{0.993}     \FloatTok{0.993}     \FloatTok{0.994}     \FloatTok{0.991}     \FloatTok{0.994}  
   \DecValTok{8}\NormalTok{ cg0000}\OperatorTok{~}\StringTok{    }\FloatTok{0.0215}   \FloatTok{0.0202}    \FloatTok{0.0187}    \FloatTok{0.0169}    \FloatTok{0.0162}    \FloatTok{0.0143} 
   \DecValTok{9}\NormalTok{ cg0000}\OperatorTok{~}\StringTok{    }\FloatTok{0.0105}   \FloatTok{0.00518}   \FloatTok{0.00410}   \FloatTok{0.00671}   \FloatTok{0.00758}   \FloatTok{0.00518}
  \DecValTok{10}\NormalTok{ cg0001}\OperatorTok{~}\StringTok{    }\FloatTok{0.634}    \FloatTok{0.635}     \FloatTok{0.621}     \FloatTok{0.639}     \FloatTok{0.599}     \FloatTok{0.591}  
  \CommentTok{# ... with 27,568 more rows, and 10 more variables: GSM946059 <dbl>,}
  \CommentTok{#   GSM946062 <dbl>, GSM946064 <dbl>, GSM946065 <dbl>, GSM946066 <dbl>,}
  \CommentTok{#   GSM946067 <dbl>, GSM946073 <dbl>, GSM946074 <dbl>, GSM946075 <dbl>,}
  \CommentTok{#   GSM946076 <dbl>}
\end{Highlighting}
\end{Shaded}

\emph{IMPORTANT NOTE}: Be sure that the first column contains the CpG names. Sometimes, your imported data look like this one (it can happen, for instance, if the \texttt{csv} file was created in R without indicating \texttt{row.names=FALSE})

\begin{verbatim}
> mydata

# A tibble: 473,999 x 6
      X1 Row.names BIB_15586_1X BIB_33043_1X EDP_5245_1X KAN_584_1X 
   <int> <chr>            <dbl>        <dbl>       <dbl>      <dbl>     
 1     1 cg000000~       0.635        0.575       0.614      0.631     
 2     2 cg000001~       0.954        0.948       0.933      0.950     
 3     3 cg000001~       0.889        0.899       0.901      0.892     
 4     4 cg000001~       0.115        0.124       0.107      0.123     
 5     5 cg000002~       0.850        0.753       0.806      0.815     
 6     6 cg000002~       0.676        0.771       0.729      0.665     
 7     7 cg000002~       0.871        0.850       0.852      0.863     
 8     8 cg000003~       0.238        0.174       0.316      0.206
\end{verbatim}

If so, the first column must be removed before being used as the input object in \texttt{DNAmAge} funcion. It can be done using \texttt{dplyr} function

\begin{verbatim}
> mydata2 <- select(mydata, -1)

# A tibble: 473,999 x 5
      Row.names BIB_15586_1X BIB_33043_1X EDP_5245_1X KAN_584_1X 
    <chr>            <dbl>        <dbl>       <dbl>      <dbl>     
 1    cg000000~       0.635        0.575       0.614      0.631     
 2    cg000001~       0.954        0.948       0.933      0.950     
 3    cg000001~       0.889        0.899       0.901      0.892     
 4    cg000001~       0.115        0.124       0.107      0.123     
 5    cg000002~       0.850        0.753       0.806      0.815     
 6    cg000002~       0.676        0.771       0.729      0.665     
 7    cg000002~       0.871        0.850       0.852      0.863     
 8    cg000003~       0.238        0.174       0.316      0.206
\end{verbatim}

In any case, if you use the object \texttt{mydata} that contains the CpGs in the second column, you will see this error message:

\begin{verbatim}
> DNAmAge(mydata)
Error in DNAmAge(mydata) : First column should contain CpG names
\end{verbatim}

DNAmAge can be estimated by simply:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{age.example55 <-}\StringTok{ }\KeywordTok{DNAmAge}\NormalTok{(MethylationData)}
\NormalTok{age.example55}
  \CommentTok{# A tibble: 16 x 7}
\NormalTok{     id        Horvath Hannum Levine   BNN skinHorvath PedBE}
     \OperatorTok{<}\NormalTok{fct}\OperatorTok{>}\StringTok{       }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{lgl}\OperatorTok{>}\StringTok{   }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{lgl}\OperatorTok{>}\StringTok{       }\ErrorTok{<}\NormalTok{lgl}\OperatorTok{>}
\StringTok{   }\DecValTok{1}\NormalTok{ GSM946048   }\FloatTok{51.8}  \OtherTok{NA}      \FloatTok{-30.3} \FloatTok{56.4}  \OtherTok{NA}          \OtherTok{NA}   
   \DecValTok{2}\NormalTok{ GSM946049   }\FloatTok{39.8}  \OtherTok{NA}      \FloatTok{-29.6} \FloatTok{42.1}  \OtherTok{NA}          \OtherTok{NA}   
   \DecValTok{3}\NormalTok{ GSM946052   }\FloatTok{26.4}  \OtherTok{NA}      \FloatTok{-33.3} \FloatTok{25.6}  \OtherTok{NA}          \OtherTok{NA}   
   \DecValTok{4}\NormalTok{ GSM946054   }\FloatTok{34.0}  \OtherTok{NA}      \FloatTok{-36.0} \FloatTok{28.0}  \OtherTok{NA}          \OtherTok{NA}   
   \DecValTok{5}\NormalTok{ GSM946055   }\FloatTok{10.1}  \OtherTok{NA}      \FloatTok{-52.8} \FloatTok{13.4}  \OtherTok{NA}          \OtherTok{NA}   
   \DecValTok{6}\NormalTok{ GSM946056   }\FloatTok{20.4}  \OtherTok{NA}      \FloatTok{-42.2} \FloatTok{16.7}  \OtherTok{NA}          \OtherTok{NA}   
   \DecValTok{7}\NormalTok{ GSM946059    }\FloatTok{6.00} \OtherTok{NA}      \FloatTok{-44.8}  \FloatTok{7.54} \OtherTok{NA}          \OtherTok{NA}   
   \DecValTok{8}\NormalTok{ GSM946062   }\FloatTok{34.6}  \OtherTok{NA}      \FloatTok{-23.2} \FloatTok{34.6}  \OtherTok{NA}          \OtherTok{NA}   
   \DecValTok{9}\NormalTok{ GSM946064    }\FloatTok{7.91} \OtherTok{NA}      \FloatTok{-49.8} \FloatTok{12.0}  \OtherTok{NA}          \OtherTok{NA}   
  \DecValTok{10}\NormalTok{ GSM946065    }\FloatTok{4.72} \OtherTok{NA}      \FloatTok{-48.2}  \FloatTok{6.43} \OtherTok{NA}          \OtherTok{NA}   
  \DecValTok{11}\NormalTok{ GSM946066   }\FloatTok{29.6}  \OtherTok{NA}      \FloatTok{-39.9} \FloatTok{28.5}  \OtherTok{NA}          \OtherTok{NA}   
  \DecValTok{12}\NormalTok{ GSM946067    }\FloatTok{1.38} \OtherTok{NA}      \FloatTok{-48.3}  \FloatTok{3.48} \OtherTok{NA}          \OtherTok{NA}   
  \DecValTok{13}\NormalTok{ GSM946073   }\FloatTok{56.0}  \OtherTok{NA}      \FloatTok{-26.7} \FloatTok{47.3}  \OtherTok{NA}          \OtherTok{NA}   
  \DecValTok{14}\NormalTok{ GSM946074   }\FloatTok{24.0}  \OtherTok{NA}      \FloatTok{-39.7} \FloatTok{23.3}  \OtherTok{NA}          \OtherTok{NA}   
  \DecValTok{15}\NormalTok{ GSM946075    }\FloatTok{9.38} \OtherTok{NA}      \FloatTok{-45.4} \FloatTok{11.9}  \OtherTok{NA}          \OtherTok{NA}   
  \DecValTok{16}\NormalTok{ GSM946076   }\FloatTok{38.8}  \OtherTok{NA}      \FloatTok{-27.5} \FloatTok{41.4}  \OtherTok{NA}          \OtherTok{NA}
\end{Highlighting}
\end{Shaded}

By default all available clocks (Hovarth, Hannum, Levine, BNN, Hovart2 and PedBE) are estimated. One may select a set of clocks by using the argument \texttt{clocks} as following:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{age.example55.sel <-}\StringTok{ }\KeywordTok{DNAmAge}\NormalTok{(MethylationData, }
                             \DataTypeTok{clocks=}\KeywordTok{c}\NormalTok{(}\StringTok{"Horvath"}\NormalTok{, }\StringTok{"BNN"}\NormalTok{))}
\NormalTok{age.example55.sel}
  \CommentTok{# A tibble: 16 x 3}
\NormalTok{     id        Horvath   BNN}
     \OperatorTok{<}\NormalTok{fct}\OperatorTok{>}\StringTok{       }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}
\StringTok{   }\DecValTok{1}\NormalTok{ GSM946048   }\FloatTok{51.8}  \FloatTok{56.4} 
   \DecValTok{2}\NormalTok{ GSM946049   }\FloatTok{39.8}  \FloatTok{42.1} 
   \DecValTok{3}\NormalTok{ GSM946052   }\FloatTok{26.4}  \FloatTok{25.6} 
   \DecValTok{4}\NormalTok{ GSM946054   }\FloatTok{34.0}  \FloatTok{28.0} 
   \DecValTok{5}\NormalTok{ GSM946055   }\FloatTok{10.1}  \FloatTok{13.4} 
   \DecValTok{6}\NormalTok{ GSM946056   }\FloatTok{20.4}  \FloatTok{16.7} 
   \DecValTok{7}\NormalTok{ GSM946059    }\FloatTok{6.00}  \FloatTok{7.54}
   \DecValTok{8}\NormalTok{ GSM946062   }\FloatTok{34.6}  \FloatTok{34.6} 
   \DecValTok{9}\NormalTok{ GSM946064    }\FloatTok{7.91} \FloatTok{12.0} 
  \DecValTok{10}\NormalTok{ GSM946065    }\FloatTok{4.72}  \FloatTok{6.43}
  \DecValTok{11}\NormalTok{ GSM946066   }\FloatTok{29.6}  \FloatTok{28.5} 
  \DecValTok{12}\NormalTok{ GSM946067    }\FloatTok{1.38}  \FloatTok{3.48}
  \DecValTok{13}\NormalTok{ GSM946073   }\FloatTok{56.0}  \FloatTok{47.3} 
  \DecValTok{14}\NormalTok{ GSM946074   }\FloatTok{24.0}  \FloatTok{23.3} 
  \DecValTok{15}\NormalTok{ GSM946075    }\FloatTok{9.38} \FloatTok{11.9} 
  \DecValTok{16}\NormalTok{ GSM946076   }\FloatTok{38.8}  \FloatTok{41.4}
\end{Highlighting}
\end{Shaded}

\hypertarget{age-acceleration}{%
\subsection{Age acceleration}\label{age-acceleration}}

However, in epidemiological studies one is intereste in assessing whether age acceleration is associated with a given trait or condition. Three different measures can be computed:

\begin{itemize}
\tightlist
\item
  \textbf{ageAcc}: Difference between DNAmAge and chronological age.
\item
  \textbf{ageAcc2}: Residuals obtained after regressing chronological age and DNAmAge (similar to IEAA).
\item
  \textbf{ageAcc3}: Residuals obtained after regressing chronological age and DNAmAge adjusted for cell counts (similar to EEAA).
\end{itemize}

All this estimates can be obtained for each clock when providing chronological age through \texttt{age} argument. This information is normally provided in a different file including different covariates (metadata or sample annotation data). In this example data are available at `SampleAnnotationExample55.csv' file that is also available at \texttt{methylclock} package:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{covariates <-}\StringTok{ }\KeywordTok{read_csv}\NormalTok{(}\KeywordTok{file.path}\NormalTok{(path, }
                                 \StringTok{"SampleAnnotationExample55.csv"}\NormalTok{))}
\NormalTok{covariates}
  \CommentTok{# A tibble: 16 x 14}
\NormalTok{     OriginalOrder id    title geo_accession TissueDetailed Tissue}
             \OperatorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{chr}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{chr}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{chr}\OperatorTok{>}\StringTok{         }\ErrorTok{<}\NormalTok{chr}\OperatorTok{>}\StringTok{          }\ErrorTok{<}\NormalTok{chr}\OperatorTok{>}\StringTok{ }
\StringTok{   }\DecValTok{1}             \DecValTok{3}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Auti}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946048     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{   }\DecValTok{2}             \DecValTok{4}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Cont}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946049     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{   }\DecValTok{3}             \DecValTok{7}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Auti}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946052     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{   }\DecValTok{4}             \DecValTok{9}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Auti}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946054     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{   }\DecValTok{5}            \DecValTok{10}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Auti}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946055     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{   }\DecValTok{6}            \DecValTok{11}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Auti}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946056     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{   }\DecValTok{7}            \DecValTok{14}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Cont}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946059     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{   }\DecValTok{8}            \DecValTok{17}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Cont}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946062     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{   }\DecValTok{9}            \DecValTok{19}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Auti}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946064     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{  }\DecValTok{10}            \DecValTok{20}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Auti}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946065     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{  }\DecValTok{11}            \DecValTok{21}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Auti}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946066     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{  }\DecValTok{12}            \DecValTok{22}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Cont}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946067     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{  }\DecValTok{13}            \DecValTok{28}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Cont}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946073     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{  }\DecValTok{14}            \DecValTok{29}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Cont}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946074     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{  }\DecValTok{15}            \DecValTok{30}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Cont}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946075     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{  }\DecValTok{16}            \DecValTok{31}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{ }\NormalTok{Cont}\OperatorTok{~}\StringTok{ }\NormalTok{GSM946076     Fresh frozen }\OperatorTok{~}\StringTok{ }\NormalTok{occip}\OperatorTok{~}
\StringTok{  }\CommentTok{# ... with 8 more variables: diseaseStatus <dbl>, Age <dbl>,}
\StringTok{  }\CommentTok{#   PostMortemInterval <dbl>, CauseofDeath <chr>, individual <dbl>,}
\StringTok{  }\CommentTok{#   Female <dbl>, Caucasian <lgl>, FemaleOriginal <lgl>}
\end{Highlighting}
\end{Shaded}

In this case, chronological age is available at \texttt{Age} column:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{age <-}\StringTok{ }\NormalTok{covariates}\OperatorTok{$}\NormalTok{Age}
\KeywordTok{head}\NormalTok{(age)}
\NormalTok{  [}\DecValTok{1}\NormalTok{] }\DecValTok{60} \DecValTok{39} \DecValTok{28} \DecValTok{39}  \DecValTok{8} \DecValTok{22}
\end{Highlighting}
\end{Shaded}

The different methylation clocks along with their age accelerated estimates can be simply computed by:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{age.example55 <-}\StringTok{ }\KeywordTok{DNAmAge}\NormalTok{(MethylationData, }\DataTypeTok{age=}\NormalTok{age, }
                         \DataTypeTok{cell.count=}\OtherTok{TRUE}\NormalTok{)}
\NormalTok{age.example55}
  \CommentTok{# A tibble: 16 x 17}
\NormalTok{     id    Horvath ageAcc.Horvath ageAcc2.Horvath ageAcc3.Horvath Hannum Levine}
     \OperatorTok{<}\NormalTok{fct}\OperatorTok{>}\StringTok{   }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{          }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{           }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{           }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{lgl}\OperatorTok{>}\StringTok{   }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}
\StringTok{   }\DecValTok{1}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{   }\FloatTok{51.8}          \FloatTok{-8.22}          \FloatTok{-4.45}            \FloatTok{-4.91}  \OtherTok{NA}      \FloatTok{-30.3}
   \DecValTok{2}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{   }\FloatTok{39.8}           \FloatTok{0.754}          \FloatTok{2.00}             \FloatTok{1.59}  \OtherTok{NA}      \FloatTok{-29.6}
   \DecValTok{3}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{   }\FloatTok{26.4}          \FloatTok{-1.59}          \FloatTok{-1.67}            \FloatTok{-1.86}  \OtherTok{NA}      \FloatTok{-33.3}
   \DecValTok{4}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{   }\FloatTok{34.0}          \FloatTok{-5.00}          \FloatTok{-3.76}            \FloatTok{-0.463} \OtherTok{NA}      \FloatTok{-36.0}
   \DecValTok{5}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{   }\FloatTok{10.1}           \FloatTok{2.06}          \FloatTok{-0.428}            \FloatTok{2.82}  \OtherTok{NA}      \FloatTok{-52.8}
   \DecValTok{6}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{   }\FloatTok{20.4}          \FloatTok{-1.61}          \FloatTok{-2.42}            \FloatTok{-2.88}  \OtherTok{NA}      \FloatTok{-42.2}
   \DecValTok{7}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{    }\FloatTok{6.00}          \FloatTok{2.00}          \FloatTok{-0.971}           \FloatTok{-0.827} \OtherTok{NA}      \FloatTok{-44.8}
   \DecValTok{8}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{   }\FloatTok{34.6}           \FloatTok{6.65}           \FloatTok{6.57}             \FloatTok{5.32}  \OtherTok{NA}      \FloatTok{-23.2}
   \DecValTok{9}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{    }\FloatTok{7.91}          \FloatTok{2.91}           \FloatTok{0.0589}          \FloatTok{-2.61}  \OtherTok{NA}      \FloatTok{-49.8}
  \DecValTok{10}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{    }\FloatTok{4.72}          \FloatTok{2.72}          \FloatTok{-0.489}            \FloatTok{1.46}  \OtherTok{NA}      \FloatTok{-48.2}
  \DecValTok{11}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{   }\FloatTok{29.6}          \FloatTok{-0.427}         \FloatTok{-0.268}           \FloatTok{-1.37}  \OtherTok{NA}      \FloatTok{-39.9}
  \DecValTok{12}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{    }\FloatTok{1.38}          \FloatTok{0.375}         \FloatTok{-2.95}            \FloatTok{-2.19}  \OtherTok{NA}      \FloatTok{-48.3}
  \DecValTok{13}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{   }\FloatTok{56.0}          \FloatTok{-4.01}          \FloatTok{-0.242}            \FloatTok{1.62}  \OtherTok{NA}      \FloatTok{-26.7}
  \DecValTok{14}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{   }\FloatTok{24.0}           \FloatTok{2.03}           \FloatTok{1.23}            \FloatTok{-0.669} \OtherTok{NA}      \FloatTok{-39.7}
  \DecValTok{15}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{    }\FloatTok{9.38}          \FloatTok{1.38}          \FloatTok{-1.11}            \FloatTok{-0.885} \OtherTok{NA}      \FloatTok{-45.4}
  \DecValTok{16}\NormalTok{ GSM9}\OperatorTok{~}\StringTok{   }\FloatTok{38.8}           \FloatTok{8.76}           \FloatTok{8.92}             \FloatTok{5.85}  \OtherTok{NA}      \FloatTok{-27.5}
  \CommentTok{# ... with 10 more variables: ageAcc.Levine <dbl>, ageAcc2.Levine <dbl>,}
  \CommentTok{#   ageAcc3.Levine <dbl>, BNN <dbl>, ageAcc.BNN <dbl>, ageAcc2.BNN <dbl>,}
  \CommentTok{#   ageAcc3.BNN <dbl>, skinHorvath <lgl>, PedBE <lgl>, age <dbl>}
\end{Highlighting}
\end{Shaded}

By default, the argument \texttt{cell.count} is set equal to TRUE and, hence, can be omitted. This implies that \texttt{ageAcc3} will be computed for all clocks. In some occassions this can be very time consuming. In such cases one can simply estimate DNAmAge, accAge and accAge2 by setting \texttt{cell.count=FALSE}. NOTE: see section 3.5 to see the reference panels available to estimate cell counts.

Then, we can investigate, for instance, whether the accelerated age is associated with Autism. In that example we will use a non-parametric test (NOTE: use t-test or linear regression for large sample sizes)

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{autism <-}\StringTok{ }\NormalTok{covariates}\OperatorTok{$}\NormalTok{diseaseStatus}
\KeywordTok{kruskal.test}\NormalTok{(age.example55}\OperatorTok{$}\NormalTok{ageAcc.Horvath }\OperatorTok{~}\StringTok{ }\NormalTok{autism)}
  
\NormalTok{    Kruskal}\OperatorTok{-}\NormalTok{Wallis rank sum test}
  
\NormalTok{  data}\OperatorTok{:}\StringTok{  }\NormalTok{age.example55}\OperatorTok{$}\NormalTok{ageAcc.Horvath by autism}
\NormalTok{  Kruskal}\OperatorTok{-}\NormalTok{Wallis chi}\OperatorTok{-}\NormalTok{squared =}\StringTok{ }\FloatTok{1.3346}\NormalTok{, df =}\StringTok{ }\DecValTok{1}\NormalTok{, p}\OperatorTok{-}\NormalTok{value =}\StringTok{ }\FloatTok{0.248}
\KeywordTok{kruskal.test}\NormalTok{(age.example55}\OperatorTok{$}\NormalTok{ageAcc2.Horvath }\OperatorTok{~}\StringTok{ }\NormalTok{autism)}
  
\NormalTok{    Kruskal}\OperatorTok{-}\NormalTok{Wallis rank sum test}
  
\NormalTok{  data}\OperatorTok{:}\StringTok{  }\NormalTok{age.example55}\OperatorTok{$}\NormalTok{ageAcc2.Horvath by autism}
\NormalTok{  Kruskal}\OperatorTok{-}\NormalTok{Wallis chi}\OperatorTok{-}\NormalTok{squared =}\StringTok{ }\FloatTok{3.1875}\NormalTok{, df =}\StringTok{ }\DecValTok{1}\NormalTok{, p}\OperatorTok{-}\NormalTok{value =}\StringTok{ }\FloatTok{0.0742}
\KeywordTok{kruskal.test}\NormalTok{(age.example55}\OperatorTok{$}\NormalTok{ageAcc3.Horvath }\OperatorTok{~}\StringTok{ }\NormalTok{autism)}
  
\NormalTok{    Kruskal}\OperatorTok{-}\NormalTok{Wallis rank sum test}
  
\NormalTok{  data}\OperatorTok{:}\StringTok{  }\NormalTok{age.example55}\OperatorTok{$}\NormalTok{ageAcc3.Horvath by autism}
\NormalTok{  Kruskal}\OperatorTok{-}\NormalTok{Wallis chi}\OperatorTok{-}\NormalTok{squared =}\StringTok{ }\FloatTok{2.8235}\NormalTok{, df =}\StringTok{ }\DecValTok{1}\NormalTok{, p}\OperatorTok{-}\NormalTok{value =}\StringTok{ }\FloatTok{0.09289}
\end{Highlighting}
\end{Shaded}

\hypertarget{chronological-age-prediction-using-expressionset-data}{%
\subsection{\texorpdfstring{Chronological age prediction using \texttt{ExpressionSet} data}{Chronological age prediction using ExpressionSet data}}\label{chronological-age-prediction-using-expressionset-data}}

One may be interested in assessing association between chronologial age and DNA methylation age or evaluating how well chronological age is predicted by DNAmAge. In order to illustrate this analysis we downloaded data from GEO corresponding to a set of healthy individuals (GEO accession number GSE58045). Data can be retrieved into R by using \texttt{GEOquery} package as an \texttt{ExpressionSet} object that can be the input of our main function.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{dd <-}\StringTok{ }\NormalTok{GEOquery}\OperatorTok{::}\KeywordTok{getGEO}\NormalTok{(}\StringTok{"GSE58045"}\NormalTok{)}
\NormalTok{gse58045 <-}\StringTok{ }\NormalTok{dd[[}\DecValTok{1}\NormalTok{]]}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gse58045}
  \KeywordTok{ExpressionSet}\NormalTok{ (storageMode}\OperatorTok{:}\StringTok{ }\NormalTok{lockedEnvironment)}
\NormalTok{  assayData}\OperatorTok{:}\StringTok{ }\DecValTok{27578}\NormalTok{ features, }\DecValTok{172}\NormalTok{ samples }
\NormalTok{    element names}\OperatorTok{:}\StringTok{ }\NormalTok{exprs }
\NormalTok{  protocolData}\OperatorTok{:}\StringTok{ }\NormalTok{none}
\NormalTok{  phenoData}
\NormalTok{    sampleNames}\OperatorTok{:}\StringTok{ }\NormalTok{GSM1399890 GSM1399891 ... }\KeywordTok{GSM1400061}\NormalTok{ (}\DecValTok{172}\NormalTok{ total)}
\NormalTok{    varLabels}\OperatorTok{:}\StringTok{ }\NormalTok{title geo_accession ... twin}\OperatorTok{:}\KeywordTok{ch1}\NormalTok{ (}\DecValTok{43}\NormalTok{ total)}
\NormalTok{    varMetadata}\OperatorTok{:}\StringTok{ }\NormalTok{labelDescription}
\NormalTok{  featureData}
\NormalTok{    featureNames}\OperatorTok{:}\StringTok{ }\NormalTok{cg00000292 cg00002426 ... }\KeywordTok{cg27665659}\NormalTok{ (}\DecValTok{27578}\NormalTok{ total)}
\NormalTok{    fvarLabels}\OperatorTok{:}\StringTok{ }\NormalTok{ID Name ... }\KeywordTok{ORF}\NormalTok{ (}\DecValTok{38}\NormalTok{ total)}
\NormalTok{    fvarMetadata}\OperatorTok{:}\StringTok{ }\NormalTok{Column Description labelDescription}
\NormalTok{  experimentData}\OperatorTok{:}\StringTok{ }\NormalTok{use }\StringTok{'experimentData(object)'}
\NormalTok{    pubMedIds}\OperatorTok{:}\StringTok{ }\DecValTok{22532803} 
\NormalTok{  Annotation}\OperatorTok{:}\StringTok{ }\NormalTok{GPL8490}
\end{Highlighting}
\end{Shaded}

The chronological age is obtained by using \texttt{pData} function from \texttt{Biobase} package that is able to deal with \texttt{ExpressionSet} objects:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(Biobase)}
\NormalTok{pheno <-}\StringTok{ }\KeywordTok{pData}\NormalTok{(gse58045)}
\NormalTok{age <-}\StringTok{ }\KeywordTok{as.numeric}\NormalTok{(pheno}\OperatorTok{$}\StringTok{`}\DataTypeTok{age:ch1}\StringTok{`}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

And the different DNA methylation age estimates are obtained by using \texttt{DNAmAge} function (NOTE: as there are missing values, the program automatically runs \texttt{impute.knn} function to get complete cases):

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{age.gse58045 <-}\StringTok{ }\KeywordTok{DNAmAge}\NormalTok{(gse58045, }\DataTypeTok{age=}\NormalTok{age)}
\NormalTok{  Imputing missing data of the entire matrix .... }
\NormalTok{  Data imputed. Starting DNAm clock estimation ...}
\NormalTok{age.gse58045}
  \CommentTok{# A tibble: 172 x 17}
\NormalTok{     id    Horvath ageAcc.Horvath ageAcc2.Horvath ageAcc3.Horvath Hannum Levine}
     \OperatorTok{<}\NormalTok{fct}\OperatorTok{>}\StringTok{   }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{          }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{           }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{           }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{lgl}\OperatorTok{>}\StringTok{   }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}
\StringTok{   }\DecValTok{1}\NormalTok{ GSM1}\OperatorTok{~}\StringTok{    }\FloatTok{65.6}          \FloatTok{1.07}            \FloatTok{4.58}            \FloatTok{5.46}  \OtherTok{NA}       \FloatTok{50.7}
   \DecValTok{2}\NormalTok{ GSM1}\OperatorTok{~}\StringTok{    }\FloatTok{66.3}          \FloatTok{0.197}           \FloatTok{4.06}            \FloatTok{5.06}  \OtherTok{NA}       \FloatTok{51.3}
   \DecValTok{3}\NormalTok{ GSM1}\OperatorTok{~}\StringTok{    }\FloatTok{53.9}         \FloatTok{-5.31}           \FloatTok{-2.98}           \FloatTok{-2.42}  \OtherTok{NA}       \FloatTok{40.5}
   \DecValTok{4}\NormalTok{ GSM1}\OperatorTok{~}\StringTok{    }\FloatTok{40.6}         \FloatTok{-5.23}           \FloatTok{-5.89}           \FloatTok{-6.14}  \OtherTok{NA}       \FloatTok{31.3}
   \DecValTok{5}\NormalTok{ GSM1}\OperatorTok{~}\StringTok{    }\FloatTok{50.1}          \FloatTok{0.982}           \FloatTok{1.06}            \FloatTok{1.28}  \OtherTok{NA}       \FloatTok{41.1}
   \DecValTok{6}\NormalTok{ GSM1}\OperatorTok{~}\StringTok{    }\FloatTok{63.7}         \FloatTok{-0.895}           \FloatTok{2.64}            \FloatTok{2.92}  \OtherTok{NA}       \FloatTok{48.1}
   \DecValTok{7}\NormalTok{ GSM1}\OperatorTok{~}\StringTok{    }\FloatTok{44.7}         \FloatTok{-0.875}          \FloatTok{-1.59}           \FloatTok{-1.76}  \OtherTok{NA}       \FloatTok{29.2}
   \DecValTok{8}\NormalTok{ GSM1}\OperatorTok{~}\StringTok{    }\FloatTok{59.7}         \FloatTok{-8.55}           \FloatTok{-4.20}           \FloatTok{-3.48}  \OtherTok{NA}       \FloatTok{41.0}
   \DecValTok{9}\NormalTok{ GSM1}\OperatorTok{~}\StringTok{    }\FloatTok{48.4}         \FloatTok{-5.84}           \FloatTok{-4.63}           \FloatTok{-2.50}  \OtherTok{NA}       \FloatTok{43.8}
  \DecValTok{10}\NormalTok{ GSM1}\OperatorTok{~}\StringTok{    }\FloatTok{59.3}         \FloatTok{-3.93}           \FloatTok{-0.719}          \FloatTok{-0.609} \OtherTok{NA}       \FloatTok{46.1}
  \CommentTok{# ... with 162 more rows, and 10 more variables: ageAcc.Levine <dbl>,}
  \CommentTok{#   ageAcc2.Levine <dbl>, ageAcc3.Levine <dbl>, BNN <dbl>, ageAcc.BNN <dbl>,}
  \CommentTok{#   ageAcc2.BNN <dbl>, ageAcc3.BNN <dbl>, skinHorvath <lgl>, PedBE <lgl>,}
  \CommentTok{#   age <dbl>}
\end{Highlighting}
\end{Shaded}

Figure \ref{fig:horvath_age} shows the correlation between DNAmAge obtained from Horvath's method and the chronological age, while Figure \ref{fig:bnn_age} depicts the correlation of a new method based on fitting a Bayesian Neural Network to predict DNAmAge based on Horvath's CpGs.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{plotDNAmAge}\NormalTok{(age.gse58045}\OperatorTok{$}\NormalTok{Horvath, age)}
\end{Highlighting}
\end{Shaded}

\begin{adjustwidth}{\fltoffset}{0mm}
\includegraphics{methylclock_files/figure-latex/horvat_age-1} \end{adjustwidth}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{plotDNAmAge}\NormalTok{(age.gse58045}\OperatorTok{$}\NormalTok{BNN, age, }\DataTypeTok{tit=}\StringTok{"Bayesian Neural Network"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{adjustwidth}{\fltoffset}{0mm}
\includegraphics{methylclock_files/figure-latex/bnn_age-1} \end{adjustwidth}

\hypertarget{use-of-dnamage-in-association-studies}{%
\subsection{Use of DNAmAge in association studies}\label{use-of-dnamage-in-association-studies}}

Let us illustrate how to use DNAmAge information in association studies (e.g case/control, smokers/non-smokers, responders/non-responders, \ldots{}). GEO number GSE58045 contains transcriptomic and epigenomic data of a study in lung cancer. Data can be retrieved into R by

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{dd <-}\StringTok{ }\NormalTok{GEOquery}\OperatorTok{::}\KeywordTok{getGEO}\NormalTok{(}\StringTok{"GSE19711"}\NormalTok{)}
\NormalTok{gse19711 <-}\StringTok{ }\NormalTok{dd[[}\DecValTok{1}\NormalTok{]]}
\end{Highlighting}
\end{Shaded}

The object \texttt{gse19711}is an \texttt{ExpressionSet} that can contains CpGs and phenotypic (e.g clinical) information

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gse19711}
  \KeywordTok{ExpressionSet}\NormalTok{ (storageMode}\OperatorTok{:}\StringTok{ }\NormalTok{lockedEnvironment)}
\NormalTok{  assayData}\OperatorTok{:}\StringTok{ }\DecValTok{27578}\NormalTok{ features, }\DecValTok{540}\NormalTok{ samples }
\NormalTok{    element names}\OperatorTok{:}\StringTok{ }\NormalTok{exprs }
\NormalTok{  protocolData}\OperatorTok{:}\StringTok{ }\NormalTok{none}
\NormalTok{  phenoData}
\NormalTok{    sampleNames}\OperatorTok{:}\StringTok{ }\NormalTok{GSM491937 GSM491938 ... }\KeywordTok{GSM492476}\NormalTok{ (}\DecValTok{540}\NormalTok{ total)}
\NormalTok{    varLabels}\OperatorTok{:}\StringTok{ }\NormalTok{title geo_accession ... stage}\OperatorTok{:}\KeywordTok{ch1}\NormalTok{ (}\DecValTok{58}\NormalTok{ total)}
\NormalTok{    varMetadata}\OperatorTok{:}\StringTok{ }\NormalTok{labelDescription}
\NormalTok{  featureData}
\NormalTok{    featureNames}\OperatorTok{:}\StringTok{ }\NormalTok{cg00000292 cg00002426 ... }\KeywordTok{cg27665659}\NormalTok{ (}\DecValTok{27578}\NormalTok{ total)}
\NormalTok{    fvarLabels}\OperatorTok{:}\StringTok{ }\NormalTok{ID Name ... }\KeywordTok{ORF}\NormalTok{ (}\DecValTok{38}\NormalTok{ total)}
\NormalTok{    fvarMetadata}\OperatorTok{:}\StringTok{ }\NormalTok{Column Description labelDescription}
\NormalTok{  experimentData}\OperatorTok{:}\StringTok{ }\NormalTok{use }\StringTok{'experimentData(object)'}
\NormalTok{    pubMedIds}\OperatorTok{:}\StringTok{ }\DecValTok{20219944} 
\NormalTok{  Annotation}\OperatorTok{:}\StringTok{ }\NormalTok{GPL8490}
\end{Highlighting}
\end{Shaded}

Let us imagine we are interested in comparing the accelerated age between cases and controls. Age and case/control status information can be obtained by:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pheno <-}\StringTok{ }\KeywordTok{pData}\NormalTok{(gse19711)}
\NormalTok{age <-}\StringTok{ }\KeywordTok{as.numeric}\NormalTok{(pheno}\OperatorTok{$}\StringTok{`}\DataTypeTok{ageatrecruitment:ch1}\StringTok{`}\NormalTok{)}
\NormalTok{disease <-}\StringTok{ }\NormalTok{pheno}\OperatorTok{$}\StringTok{`}\DataTypeTok{sample type:ch1}\StringTok{`}
\KeywordTok{table}\NormalTok{(disease)}
\NormalTok{  disease}
\NormalTok{     bi}\OperatorTok{-}\NormalTok{sulphite converted genomic whole blood DNA from Case }
                                                         \DecValTok{266} 
\NormalTok{  bi}\OperatorTok{-}\NormalTok{sulphite converted genomic whole blood DNA from Control }
                                                         \DecValTok{274}
\NormalTok{disease[}\KeywordTok{grep}\NormalTok{(}\StringTok{"Control"}\NormalTok{, disease)] <-}\StringTok{ "Control"}
\NormalTok{disease[}\KeywordTok{grep}\NormalTok{(}\StringTok{"Case"}\NormalTok{, disease)] <-}\StringTok{ "Case"}
\NormalTok{disease <-}\StringTok{ }\KeywordTok{factor}\NormalTok{(disease, }\DataTypeTok{levels=}\KeywordTok{c}\NormalTok{(}\StringTok{"Control"}\NormalTok{, }\StringTok{"Case"}\NormalTok{))}
\KeywordTok{table}\NormalTok{(disease)}
\NormalTok{  disease}
\NormalTok{  Control    Case }
      \DecValTok{274}     \DecValTok{266}
\end{Highlighting}
\end{Shaded}

The DNAmAge estimates of different methods is computed by

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{age.gse19711 <-}\StringTok{ }\KeywordTok{DNAmAge}\NormalTok{(gse19711, }\DataTypeTok{age=}\NormalTok{age)}
\NormalTok{  Imputing missing data of the entire matrix .... }
\NormalTok{  Data imputed. Starting DNAm clock estimation ...}
\end{Highlighting}
\end{Shaded}

We can observe there are missing data. The funcion automatically impute those using \texttt{impute.knn} function from \texttt{impute} package since complete cases are required to compute the different methylation clocks. The estimates are:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{age.gse19711}
  \CommentTok{# A tibble: 540 x 17}
\NormalTok{     id    Horvath ageAcc.Horvath ageAcc2.Horvath ageAcc3.Horvath Hannum Levine}
     \OperatorTok{<}\NormalTok{fct}\OperatorTok{>}\StringTok{   }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{          }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{           }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{           }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{lgl}\OperatorTok{>}\StringTok{   }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}
\StringTok{   }\DecValTok{1}\NormalTok{ GSM4}\OperatorTok{~}\StringTok{    }\FloatTok{62.9}          \FloatTok{-5.14}          \FloatTok{-0.351}          \FloatTok{-1.10}  \OtherTok{NA}       \FloatTok{61.1}
   \DecValTok{2}\NormalTok{ GSM4}\OperatorTok{~}\StringTok{    }\FloatTok{68.8}         \FloatTok{-12.2}           \FloatTok{-2.85}           \FloatTok{-2.13}  \OtherTok{NA}       \FloatTok{57.0}
   \DecValTok{3}\NormalTok{ GSM4}\OperatorTok{~}\StringTok{    }\FloatTok{60.0}           \FloatTok{3.96}           \FloatTok{4.54}            \FloatTok{4.37}  \OtherTok{NA}       \FloatTok{43.0}
   \DecValTok{4}\NormalTok{ GSM4}\OperatorTok{~}\StringTok{    }\FloatTok{57.9}          \FloatTok{-4.13}          \FloatTok{-1.45}           \FloatTok{-1.38}  \OtherTok{NA}       \FloatTok{40.9}
   \DecValTok{5}\NormalTok{ GSM4}\OperatorTok{~}\StringTok{    }\FloatTok{59.0}         \FloatTok{-13.0}           \FloatTok{-6.79}           \FloatTok{-6.98}  \OtherTok{NA}       \FloatTok{57.0}
   \DecValTok{6}\NormalTok{ GSM4}\OperatorTok{~}\StringTok{    }\FloatTok{57.0}          \FloatTok{-4.00}          \FloatTok{-1.66}           \FloatTok{-1.09}  \OtherTok{NA}       \FloatTok{44.7}
   \DecValTok{7}\NormalTok{ GSM4}\OperatorTok{~}\StringTok{    }\FloatTok{61.9}          \FloatTok{-3.08}           \FloatTok{0.657}           \FloatTok{0.183} \OtherTok{NA}       \FloatTok{47.9}
   \DecValTok{8}\NormalTok{ GSM4}\OperatorTok{~}\StringTok{    }\FloatTok{59.1}         \FloatTok{-11.9}           \FloatTok{-6.07}           \FloatTok{-5.53}  \OtherTok{NA}       \FloatTok{50.0}
   \DecValTok{9}\NormalTok{ GSM4}\OperatorTok{~}\StringTok{    }\FloatTok{60.7}         \FloatTok{-16.3}           \FloatTok{-8.33}           \FloatTok{-9.33}  \OtherTok{NA}       \FloatTok{47.7}
  \DecValTok{10}\NormalTok{ GSM4}\OperatorTok{~}\StringTok{    }\FloatTok{51.1}          \FloatTok{-7.93}          \FloatTok{-6.30}           \FloatTok{-6.33}  \OtherTok{NA}       \FloatTok{52.5}
  \CommentTok{# ... with 530 more rows, and 10 more variables: ageAcc.Levine <dbl>,}
  \CommentTok{#   ageAcc2.Levine <dbl>, ageAcc3.Levine <dbl>, BNN <dbl>, ageAcc.BNN <dbl>,}
  \CommentTok{#   ageAcc2.BNN <dbl>, ageAcc3.BNN <dbl>, skinHorvath <lgl>, PedBE <lgl>,}
  \CommentTok{#   age <dbl>}
\end{Highlighting}
\end{Shaded}

The association between disease status and DNAmAge estimated using Horvath's method can be computed by

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{mod.horvath1 <-}\StringTok{ }\KeywordTok{glm}\NormalTok{(disease }\OperatorTok{~}\StringTok{ }\NormalTok{ageAcc.Horvath , }
                   \DataTypeTok{data=}\NormalTok{age.gse19711,}
                   \DataTypeTok{family=}\StringTok{"binomial"}\NormalTok{)}
\KeywordTok{summary}\NormalTok{(mod.horvath1)}
  
\NormalTok{  Call}\OperatorTok{:}
\StringTok{  }\KeywordTok{glm}\NormalTok{(}\DataTypeTok{formula =}\NormalTok{ disease }\OperatorTok{~}\StringTok{ }\NormalTok{ageAcc.Horvath, }\DataTypeTok{family =} \StringTok{"binomial"}\NormalTok{, }
      \DataTypeTok{data =}\NormalTok{ age.gse19711)}
  
\NormalTok{  Deviance Residuals}\OperatorTok{:}\StringTok{ }
\StringTok{     }\NormalTok{Min      1Q  Median      3Q     Max  }
  \FloatTok{-1.358}  \FloatTok{-1.160}  \FloatTok{-1.030}   \FloatTok{1.184}   \FloatTok{1.771}  
  
\NormalTok{  Coefficients}\OperatorTok{:}
\StringTok{                 }\NormalTok{Estimate Std. Error z value }\KeywordTok{Pr}\NormalTok{(}\OperatorTok{>}\ErrorTok{|}\NormalTok{z}\OperatorTok{|}\NormalTok{)  }
\NormalTok{  (Intercept)    }\FloatTok{-0.10995}    \FloatTok{0.09771}  \FloatTok{-1.125}   \FloatTok{0.2605}  
\NormalTok{  ageAcc.Horvath }\FloatTok{-0.02023}    \FloatTok{0.01154}  \FloatTok{-1.753}   \FloatTok{0.0795}\NormalTok{ .}
  \OperatorTok{---}
\StringTok{  }\NormalTok{Signif. codes}\OperatorTok{:}\StringTok{  }\DecValTok{0} \StringTok{'***'} \FloatTok{0.001} \StringTok{'**'} \FloatTok{0.01} \StringTok{'*'} \FloatTok{0.05} \StringTok{'.'} \FloatTok{0.1} \StringTok{' '} \DecValTok{1}
  
\NormalTok{  (Dispersion parameter }\ControlFlowTok{for}\NormalTok{ binomial family taken to be }\DecValTok{1}\NormalTok{)}
  
\NormalTok{      Null deviance}\OperatorTok{:}\StringTok{ }\FloatTok{748.48}\NormalTok{  on }\DecValTok{539}\NormalTok{  degrees of freedom}
\NormalTok{  Residual deviance}\OperatorTok{:}\StringTok{ }\FloatTok{745.25}\NormalTok{  on }\DecValTok{538}\NormalTok{  degrees of freedom}
\NormalTok{  AIC}\OperatorTok{:}\StringTok{ }\FloatTok{749.25}
  
\NormalTok{  Number of Fisher Scoring iterations}\OperatorTok{:}\StringTok{ }\DecValTok{4}

\NormalTok{mod.horvath2 <-}\StringTok{ }\KeywordTok{glm}\NormalTok{(disease }\OperatorTok{~}\StringTok{ }\NormalTok{ageAcc2.Horvath , }
                   \DataTypeTok{data=}\NormalTok{age.gse19711,}
                   \DataTypeTok{family=}\StringTok{"binomial"}\NormalTok{)}
\KeywordTok{summary}\NormalTok{(mod.horvath2)}
  
\NormalTok{  Call}\OperatorTok{:}
\StringTok{  }\KeywordTok{glm}\NormalTok{(}\DataTypeTok{formula =}\NormalTok{ disease }\OperatorTok{~}\StringTok{ }\NormalTok{ageAcc2.Horvath, }\DataTypeTok{family =} \StringTok{"binomial"}\NormalTok{, }
      \DataTypeTok{data =}\NormalTok{ age.gse19711)}
  
\NormalTok{  Deviance Residuals}\OperatorTok{:}\StringTok{ }
\StringTok{     }\NormalTok{Min      1Q  Median      3Q     Max  }
  \FloatTok{-1.279}  \FloatTok{-1.163}  \FloatTok{-1.082}   \FloatTok{1.189}   \FloatTok{1.589}  
  
\NormalTok{  Coefficients}\OperatorTok{:}
\StringTok{                  }\NormalTok{Estimate Std. Error z value }\KeywordTok{Pr}\NormalTok{(}\OperatorTok{>}\ErrorTok{|}\NormalTok{z}\OperatorTok{|}\NormalTok{)}
\NormalTok{  (Intercept)     }\FloatTok{-0.02970}    \FloatTok{0.08617}  \FloatTok{-0.345}    \FloatTok{0.730}
\NormalTok{  ageAcc2.Horvath }\FloatTok{-0.01315}    \FloatTok{0.01209}  \FloatTok{-1.087}    \FloatTok{0.277}
  
\NormalTok{  (Dispersion parameter }\ControlFlowTok{for}\NormalTok{ binomial family taken to be }\DecValTok{1}\NormalTok{)}
  
\NormalTok{      Null deviance}\OperatorTok{:}\StringTok{ }\FloatTok{748.48}\NormalTok{  on }\DecValTok{539}\NormalTok{  degrees of freedom}
\NormalTok{  Residual deviance}\OperatorTok{:}\StringTok{ }\FloatTok{747.27}\NormalTok{  on }\DecValTok{538}\NormalTok{  degrees of freedom}
\NormalTok{  AIC}\OperatorTok{:}\StringTok{ }\FloatTok{751.27}
  
\NormalTok{  Number of Fisher Scoring iterations}\OperatorTok{:}\StringTok{ }\DecValTok{3}

\NormalTok{mod.horvath3 <-}\StringTok{ }\KeywordTok{glm}\NormalTok{(disease }\OperatorTok{~}\StringTok{ }\NormalTok{ageAcc3.Horvath , }
                   \DataTypeTok{data=}\NormalTok{age.gse19711,}
                   \DataTypeTok{family=}\StringTok{"binomial"}\NormalTok{)}
\KeywordTok{summary}\NormalTok{(mod.horvath3)}
  
\NormalTok{  Call}\OperatorTok{:}
\StringTok{  }\KeywordTok{glm}\NormalTok{(}\DataTypeTok{formula =}\NormalTok{ disease }\OperatorTok{~}\StringTok{ }\NormalTok{ageAcc3.Horvath, }\DataTypeTok{family =} \StringTok{"binomial"}\NormalTok{, }
      \DataTypeTok{data =}\NormalTok{ age.gse19711)}
  
\NormalTok{  Deviance Residuals}\OperatorTok{:}\StringTok{ }
\StringTok{     }\NormalTok{Min      1Q  Median      3Q     Max  }
  \FloatTok{-1.338}  \FloatTok{-1.163}  \FloatTok{-1.046}   \FloatTok{1.185}   \FloatTok{1.771}  
  
\NormalTok{  Coefficients}\OperatorTok{:}
\StringTok{                  }\NormalTok{Estimate Std. Error z value }\KeywordTok{Pr}\NormalTok{(}\OperatorTok{>}\ErrorTok{|}\NormalTok{z}\OperatorTok{|}\NormalTok{)}
\NormalTok{  (Intercept)     }\FloatTok{-0.02993}    \FloatTok{0.08626}  \FloatTok{-0.347}    \FloatTok{0.729}
\NormalTok{  ageAcc3.Horvath }\FloatTok{-0.01927}    \FloatTok{0.01283}  \FloatTok{-1.502}    \FloatTok{0.133}
  
\NormalTok{  (Dispersion parameter }\ControlFlowTok{for}\NormalTok{ binomial family taken to be }\DecValTok{1}\NormalTok{)}
  
\NormalTok{      Null deviance}\OperatorTok{:}\StringTok{ }\FloatTok{748.48}\NormalTok{  on }\DecValTok{539}\NormalTok{  degrees of freedom}
\NormalTok{  Residual deviance}\OperatorTok{:}\StringTok{ }\FloatTok{746.13}\NormalTok{  on }\DecValTok{538}\NormalTok{  degrees of freedom}
\NormalTok{  AIC}\OperatorTok{:}\StringTok{ }\FloatTok{750.13}
  
\NormalTok{  Number of Fisher Scoring iterations}\OperatorTok{:}\StringTok{ }\DecValTok{4}
\end{Highlighting}
\end{Shaded}

We do not observe statistical significant association between age acceleration estimated using Horvath method and the risk of developing lung cancer. It is worth to notice that Horvath's clock was created to predict chronological age and the impact of age acceleration of this clock on disease may be limited. On the other hand, Levine's clock aimed to distinguish risk between same-aged individuals. Let us evaluate whether this age acceleration usin Levine's clock is associated with lung cancer

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{mod.levine1 <-}\StringTok{ }\KeywordTok{glm}\NormalTok{(disease }\OperatorTok{~}\StringTok{ }\NormalTok{ageAcc.Levine , }\DataTypeTok{data=}\NormalTok{age.gse19711,}
           \DataTypeTok{family=}\StringTok{"binomial"}\NormalTok{)}
\KeywordTok{summary}\NormalTok{(mod.levine1)}
  
\NormalTok{  Call}\OperatorTok{:}
\StringTok{  }\KeywordTok{glm}\NormalTok{(}\DataTypeTok{formula =}\NormalTok{ disease }\OperatorTok{~}\StringTok{ }\NormalTok{ageAcc.Levine, }\DataTypeTok{family =} \StringTok{"binomial"}\NormalTok{, }\DataTypeTok{data =}\NormalTok{ age.gse19711)}
  
\NormalTok{  Deviance Residuals}\OperatorTok{:}\StringTok{ }
\StringTok{     }\NormalTok{Min      1Q  Median      3Q     Max  }
  \FloatTok{-1.592}  \FloatTok{-1.149}  \FloatTok{-0.939}   \FloatTok{1.174}   \FloatTok{1.733}  
  
\NormalTok{  Coefficients}\OperatorTok{:}
\StringTok{                }\NormalTok{Estimate Std. Error z value }\KeywordTok{Pr}\NormalTok{(}\OperatorTok{>}\ErrorTok{|}\NormalTok{z}\OperatorTok{|}\NormalTok{)   }
\NormalTok{  (Intercept)    }\FloatTok{0.40956}    \FloatTok{0.17894}   \FloatTok{2.289}  \FloatTok{0.02209} \OperatorTok{*}\StringTok{ }
\StringTok{  }\NormalTok{ageAcc.Levine  }\FloatTok{0.03178}    \FloatTok{0.01133}   \FloatTok{2.806}  \FloatTok{0.00502} \OperatorTok{**}
\StringTok{  }\OperatorTok{---}
\StringTok{  }\NormalTok{Signif. codes}\OperatorTok{:}\StringTok{  }\DecValTok{0} \StringTok{'***'} \FloatTok{0.001} \StringTok{'**'} \FloatTok{0.01} \StringTok{'*'} \FloatTok{0.05} \StringTok{'.'} \FloatTok{0.1} \StringTok{' '} \DecValTok{1}
  
\NormalTok{  (Dispersion parameter }\ControlFlowTok{for}\NormalTok{ binomial family taken to be }\DecValTok{1}\NormalTok{)}
  
\NormalTok{      Null deviance}\OperatorTok{:}\StringTok{ }\FloatTok{748.48}\NormalTok{  on }\DecValTok{539}\NormalTok{  degrees of freedom}
\NormalTok{  Residual deviance}\OperatorTok{:}\StringTok{ }\FloatTok{740.17}\NormalTok{  on }\DecValTok{538}\NormalTok{  degrees of freedom}
\NormalTok{  AIC}\OperatorTok{:}\StringTok{ }\FloatTok{744.17}
  
\NormalTok{  Number of Fisher Scoring iterations}\OperatorTok{:}\StringTok{ }\DecValTok{4}

\NormalTok{mod.levine2 <-}\StringTok{ }\KeywordTok{glm}\NormalTok{(disease }\OperatorTok{~}\StringTok{ }\NormalTok{ageAcc2.Levine , }\DataTypeTok{data=}\NormalTok{age.gse19711,}
           \DataTypeTok{family=}\StringTok{"binomial"}\NormalTok{)}
\KeywordTok{summary}\NormalTok{(mod.levine2)}
  
\NormalTok{  Call}\OperatorTok{:}
\StringTok{  }\KeywordTok{glm}\NormalTok{(}\DataTypeTok{formula =}\NormalTok{ disease }\OperatorTok{~}\StringTok{ }\NormalTok{ageAcc2.Levine, }\DataTypeTok{family =} \StringTok{"binomial"}\NormalTok{, }
      \DataTypeTok{data =}\NormalTok{ age.gse19711)}
  
\NormalTok{  Deviance Residuals}\OperatorTok{:}\StringTok{ }
\StringTok{      }\NormalTok{Min       1Q   Median       3Q      Max  }
  \FloatTok{-1.7053}  \FloatTok{-1.1328}  \FloatTok{-0.8614}   \FloatTok{1.1529}   \FloatTok{1.8015}  
  
\NormalTok{  Coefficients}\OperatorTok{:}
\StringTok{                 }\NormalTok{Estimate Std. Error z value }\KeywordTok{Pr}\NormalTok{(}\OperatorTok{>}\ErrorTok{|}\NormalTok{z}\OperatorTok{|}\NormalTok{)    }
\NormalTok{  (Intercept)    }\FloatTok{-0.02925}    \FloatTok{0.08718}  \FloatTok{-0.336} \FloatTok{0.737225}    
\NormalTok{  ageAcc2.Levine  }\FloatTok{0.04430}    \FloatTok{0.01234}   \FloatTok{3.589} \FloatTok{0.000332} \OperatorTok{**}\ErrorTok{*}
\StringTok{  }\OperatorTok{---}
\StringTok{  }\NormalTok{Signif. codes}\OperatorTok{:}\StringTok{  }\DecValTok{0} \StringTok{'***'} \FloatTok{0.001} \StringTok{'**'} \FloatTok{0.01} \StringTok{'*'} \FloatTok{0.05} \StringTok{'.'} \FloatTok{0.1} \StringTok{' '} \DecValTok{1}
  
\NormalTok{  (Dispersion parameter }\ControlFlowTok{for}\NormalTok{ binomial family taken to be }\DecValTok{1}\NormalTok{)}
  
\NormalTok{      Null deviance}\OperatorTok{:}\StringTok{ }\FloatTok{748.48}\NormalTok{  on }\DecValTok{539}\NormalTok{  degrees of freedom}
\NormalTok{  Residual deviance}\OperatorTok{:}\StringTok{ }\FloatTok{734.49}\NormalTok{  on }\DecValTok{538}\NormalTok{  degrees of freedom}
\NormalTok{  AIC}\OperatorTok{:}\StringTok{ }\FloatTok{738.49}
  
\NormalTok{  Number of Fisher Scoring iterations}\OperatorTok{:}\StringTok{ }\DecValTok{4}

\NormalTok{mod.levine3 <-}\StringTok{ }\KeywordTok{glm}\NormalTok{(disease }\OperatorTok{~}\StringTok{ }\NormalTok{ageAcc3.Levine , }\DataTypeTok{data=}\NormalTok{age.gse19711,}
           \DataTypeTok{family=}\StringTok{"binomial"}\NormalTok{)}
\KeywordTok{summary}\NormalTok{(mod.levine3)}
  
\NormalTok{  Call}\OperatorTok{:}
\StringTok{  }\KeywordTok{glm}\NormalTok{(}\DataTypeTok{formula =}\NormalTok{ disease }\OperatorTok{~}\StringTok{ }\NormalTok{ageAcc3.Levine, }\DataTypeTok{family =} \StringTok{"binomial"}\NormalTok{, }
      \DataTypeTok{data =}\NormalTok{ age.gse19711)}
  
\NormalTok{  Deviance Residuals}\OperatorTok{:}\StringTok{ }
\StringTok{     }\NormalTok{Min      1Q  Median      3Q     Max  }
  \FloatTok{-1.354}  \FloatTok{-1.161}  \FloatTok{-1.057}   \FloatTok{1.187}   \FloatTok{1.408}  
  
\NormalTok{  Coefficients}\OperatorTok{:}
\StringTok{                 }\NormalTok{Estimate Std. Error z value }\KeywordTok{Pr}\NormalTok{(}\OperatorTok{>}\ErrorTok{|}\NormalTok{z}\OperatorTok{|}\NormalTok{)}
\NormalTok{  (Intercept)    }\FloatTok{-0.02962}    \FloatTok{0.08622}  \FloatTok{-0.344}    \FloatTok{0.731}
\NormalTok{  ageAcc3.Levine  }\FloatTok{0.01679}    \FloatTok{0.01244}   \FloatTok{1.350}    \FloatTok{0.177}
  
\NormalTok{  (Dispersion parameter }\ControlFlowTok{for}\NormalTok{ binomial family taken to be }\DecValTok{1}\NormalTok{)}
  
\NormalTok{      Null deviance}\OperatorTok{:}\StringTok{ }\FloatTok{748.48}\NormalTok{  on }\DecValTok{539}\NormalTok{  degrees of freedom}
\NormalTok{  Residual deviance}\OperatorTok{:}\StringTok{ }\FloatTok{746.62}\NormalTok{  on }\DecValTok{538}\NormalTok{  degrees of freedom}
\NormalTok{  AIC}\OperatorTok{:}\StringTok{ }\FloatTok{750.62}
  
\NormalTok{  Number of Fisher Scoring iterations}\OperatorTok{:}\StringTok{ }\DecValTok{3}
\end{Highlighting}
\end{Shaded}

Here we observe as the risk of developing lung cancer increases 3.23 percent per each unit in the age accelerated variable (\texttt{ageAcc}). Similar conclusion is obtained when using \texttt{ageAcc2} and \texttt{ageAcc3} variables.

In some occasions cell composition should be used to assess association. This information is calculated in \texttt{DNAmAge} function and it can be incorporated in the model by:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{cell <-}\StringTok{ }\KeywordTok{attr}\NormalTok{(age.gse19711, }\StringTok{"cell_proportion"}\NormalTok{)}
\NormalTok{mod.cell <-}\StringTok{ }\KeywordTok{glm}\NormalTok{(disease }\OperatorTok{~}\StringTok{ }\NormalTok{ageAcc.Levine }\OperatorTok{+}\StringTok{ }\NormalTok{cell, }\DataTypeTok{data=}\NormalTok{age.gse19711,}
           \DataTypeTok{family=}\StringTok{"binomial"}\NormalTok{)}
\KeywordTok{summary}\NormalTok{(mod.cell)}
  
\NormalTok{  Call}\OperatorTok{:}
\StringTok{  }\KeywordTok{glm}\NormalTok{(}\DataTypeTok{formula =}\NormalTok{ disease }\OperatorTok{~}\StringTok{ }\NormalTok{ageAcc.Levine }\OperatorTok{+}\StringTok{ }\NormalTok{cell, }\DataTypeTok{family =} \StringTok{"binomial"}\NormalTok{, }
      \DataTypeTok{data =}\NormalTok{ age.gse19711)}
  
\NormalTok{  Deviance Residuals}\OperatorTok{:}\StringTok{ }
\StringTok{      }\NormalTok{Min       1Q   Median       3Q      Max  }
  \FloatTok{-1.9605}  \FloatTok{-1.0832}  \FloatTok{-0.6241}   \FloatTok{1.0742}   \FloatTok{2.3395}  
  
\NormalTok{  Coefficients}\OperatorTok{:}
\StringTok{                 }\NormalTok{Estimate Std. Error z value }\KeywordTok{Pr}\NormalTok{(}\OperatorTok{>}\ErrorTok{|}\NormalTok{z}\OperatorTok{|}\NormalTok{)    }
\NormalTok{  (Intercept)   }\FloatTok{-9.768206}   \FloatTok{4.380382}  \FloatTok{-2.230} \FloatTok{0.025748} \OperatorTok{*}\StringTok{  }
\StringTok{  }\NormalTok{ageAcc.Levine  }\FloatTok{0.003959}   \FloatTok{0.012208}   \FloatTok{0.324} \FloatTok{0.745746}    
\NormalTok{  cellCD4T      }\FloatTok{-3.339693}   \FloatTok{3.833531}  \FloatTok{-0.871} \FloatTok{0.383656}    
\NormalTok{  cellMono      }\FloatTok{10.165096}   \FloatTok{4.594096}   \FloatTok{2.213} \FloatTok{0.026922} \OperatorTok{*}\StringTok{  }
\StringTok{  }\NormalTok{cellNeu       }\FloatTok{16.319534}   \FloatTok{4.584745}   \FloatTok{3.560} \FloatTok{0.000372} \OperatorTok{**}\ErrorTok{*}
\StringTok{  }\NormalTok{cellNK        }\FloatTok{-0.882134}   \FloatTok{4.296498}  \FloatTok{-0.205} \FloatTok{0.837326}    
  \OperatorTok{---}
\StringTok{  }\NormalTok{Signif. codes}\OperatorTok{:}\StringTok{  }\DecValTok{0} \StringTok{'***'} \FloatTok{0.001} \StringTok{'**'} \FloatTok{0.01} \StringTok{'*'} \FloatTok{0.05} \StringTok{'.'} \FloatTok{0.1} \StringTok{' '} \DecValTok{1}
  
\NormalTok{  (Dispersion parameter }\ControlFlowTok{for}\NormalTok{ binomial family taken to be }\DecValTok{1}\NormalTok{)}
  
\NormalTok{      Null deviance}\OperatorTok{:}\StringTok{ }\FloatTok{748.48}\NormalTok{  on }\DecValTok{539}\NormalTok{  degrees of freedom}
\NormalTok{  Residual deviance}\OperatorTok{:}\StringTok{ }\FloatTok{686.56}\NormalTok{  on }\DecValTok{534}\NormalTok{  degrees of freedom}
\NormalTok{  AIC}\OperatorTok{:}\StringTok{ }\FloatTok{698.56}
  
\NormalTok{  Number of Fisher Scoring iterations}\OperatorTok{:}\StringTok{ }\DecValTok{4}
\end{Highlighting}
\end{Shaded}

Here we observe as the positive association disapears after adjusting for cell counts.

\hypertarget{gestational-dnam-age-estimation}{%
\section{Gestational DNAm Age estimation}\label{gestational-dnam-age-estimation}}

Let us start by reproducing the example provided in Knight et al. (2016) as a test data set (file `TestDataset.csv'). It consists on 3 individuals whose methylation data are available as supplementary data of their paper. The data is also available at \texttt{methylclock} package as a data frame.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{TestDataset[}\DecValTok{1}\OperatorTok{:}\DecValTok{5}\NormalTok{,]}
\NormalTok{       CpGName    Sample1    Sample2    Sample3}
  \DecValTok{1}\NormalTok{ cg00000292 }\FloatTok{0.72546496} \FloatTok{0.72350947} \FloatTok{0.69023377}
  \DecValTok{2}\NormalTok{ cg00002426 }\FloatTok{0.85091763} \FloatTok{0.80077888} \FloatTok{0.80385777}
  \DecValTok{3}\NormalTok{ cg00003994 }\FloatTok{0.05125853} \FloatTok{0.05943935} \FloatTok{0.05559333}
  \DecValTok{4}\NormalTok{ cg00005847 }\FloatTok{0.08775420} \FloatTok{0.11722333} \FloatTok{0.10845113}
  \DecValTok{5}\NormalTok{ cg00006414 }\FloatTok{0.03982478} \FloatTok{0.06146891} \FloatTok{0.03491992}
\end{Highlighting}
\end{Shaded}

The Gestational Age (in months) is simply computed by

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{ga.test <-}\StringTok{ }\KeywordTok{DNAmGA}\NormalTok{(TestDataset)}
\NormalTok{ga.test}
  \CommentTok{# A tibble: 3 x 5}
\NormalTok{    id      Knight Bohlin Mayne Lee  }
    \OperatorTok{<}\NormalTok{fct}\OperatorTok{>}\StringTok{    }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{lgl}\OperatorTok{>}\StringTok{  }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{lgl}\OperatorTok{>}
\StringTok{  }\DecValTok{1}\NormalTok{ Sample1   }\FloatTok{38.2} \OtherTok{NA}      \FloatTok{35.8} \OtherTok{NA}   
  \DecValTok{2}\NormalTok{ Sample2   }\FloatTok{38.8} \OtherTok{NA}      \FloatTok{36.5} \OtherTok{NA}   
  \DecValTok{3}\NormalTok{ Sample3   }\FloatTok{40.0} \OtherTok{NA}      \FloatTok{36.6} \OtherTok{NA}
\end{Highlighting}
\end{Shaded}

The results are the same as those described in the additional file 7 of Knight et al. (2016) (link \href{https://static-content.springer.com/esm/art\%3A10.1186\%2Fs13059-016-1068-z/MediaObjects/13059_2016_1068_MOESM7_ESM.docx}{here})

Let us continue by illustrating how to compute GA of real examples. The PROGRESS cohort data is available in the additional file 8 of Knight et al. (2016). It is availabel at \texttt{methylclock} as a \texttt{tibble}:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{progress_data}
  \CommentTok{# A tibble: 148 x 151}
\NormalTok{     CpGmarker }\StringTok{`}\DataTypeTok{784}\StringTok{`} \StringTok{`}\DataTypeTok{1052}\StringTok{`} \StringTok{`}\DataTypeTok{1048}\StringTok{`} \StringTok{`}\DataTypeTok{1017}\StringTok{`} \StringTok{`}\DataTypeTok{956}\StringTok{`} \StringTok{`}\DataTypeTok{1038}\StringTok{`} \StringTok{`}\DataTypeTok{989}\StringTok{`} \StringTok{`}\DataTypeTok{946}\StringTok{`} \StringTok{`}\DataTypeTok{941}\StringTok{`} \StringTok{`}\DataTypeTok{1024}\StringTok{`}
     \OperatorTok{<}\NormalTok{chr}\OperatorTok{>}\StringTok{     }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{  }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{  }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{  }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{  }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{  }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}
\StringTok{   }\DecValTok{1}\NormalTok{ cg000228}\OperatorTok{~}\StringTok{ }\FloatTok{0.289}  \FloatTok{0.372}  \FloatTok{0.347}  \FloatTok{0.351} \FloatTok{0.313}  \FloatTok{0.300} \FloatTok{0.298} \FloatTok{0.294} \FloatTok{0.322}  \FloatTok{0.313}
   \DecValTok{2}\NormalTok{ cg004662}\OperatorTok{~}\StringTok{ }\FloatTok{0.658}  \FloatTok{0.724}  \FloatTok{0.700}  \FloatTok{0.717} \FloatTok{0.695}  \FloatTok{0.665} \FloatTok{0.710} \FloatTok{0.686} \FloatTok{0.692}  \FloatTok{0.704}
   \DecValTok{3}\NormalTok{ cg005468}\OperatorTok{~}\StringTok{ }\FloatTok{0.682}  \FloatTok{0.711}  \FloatTok{0.684}  \FloatTok{0.717} \FloatTok{0.627}  \FloatTok{0.605} \FloatTok{0.684} \FloatTok{0.716} \FloatTok{0.684}  \FloatTok{0.666}
   \DecValTok{4}\NormalTok{ cg005757}\OperatorTok{~}\StringTok{ }\FloatTok{0.312}  \FloatTok{0.381}  \FloatTok{0.300}  \FloatTok{0.331} \FloatTok{0.294}  \FloatTok{0.348} \FloatTok{0.284} \FloatTok{0.305} \FloatTok{0.319}  \FloatTok{0.325}
   \DecValTok{5}\NormalTok{ cg006893}\OperatorTok{~}\StringTok{ }\FloatTok{0.566}  \FloatTok{0.576}  \FloatTok{0.556}  \FloatTok{0.571} \FloatTok{0.521}  \FloatTok{0.569} \FloatTok{0.599} \FloatTok{0.575} \FloatTok{0.532}  \FloatTok{0.564}
   \DecValTok{6}\NormalTok{ cg010565}\OperatorTok{~}\StringTok{ }\FloatTok{0.558}  \FloatTok{0.620}  \FloatTok{0.529}  \FloatTok{0.600} \FloatTok{0.577}  \FloatTok{0.574} \FloatTok{0.590} \FloatTok{0.576} \FloatTok{0.548}  \FloatTok{0.555}
   \DecValTok{7}\NormalTok{ cg011844}\OperatorTok{~}\StringTok{ }\FloatTok{0.712}  \FloatTok{0.718}  \FloatTok{0.667}  \FloatTok{0.744} \FloatTok{0.668}  \FloatTok{0.676} \FloatTok{0.710} \FloatTok{0.744} \FloatTok{0.685}  \FloatTok{0.717}
   \DecValTok{8}\NormalTok{ cg013480}\OperatorTok{~}\StringTok{ }\FloatTok{0.195}  \FloatTok{0.186}  \FloatTok{0.180}  \FloatTok{0.194} \FloatTok{0.212}  \FloatTok{0.208} \FloatTok{0.183} \FloatTok{0.129} \FloatTok{0.161}  \FloatTok{0.144}
   \DecValTok{9}\NormalTok{ cg021006}\OperatorTok{~}\StringTok{ }\FloatTok{0.329}  \FloatTok{0.330}  \FloatTok{0.340}  \FloatTok{0.344} \FloatTok{0.268}  \FloatTok{0.280} \FloatTok{0.288} \FloatTok{0.314} \FloatTok{0.283}  \FloatTok{0.346}
  \DecValTok{10}\NormalTok{ cg028138}\OperatorTok{~}\StringTok{ }\FloatTok{0.819}  \FloatTok{0.858}  \FloatTok{0.832}  \FloatTok{0.874} \FloatTok{0.861}  \FloatTok{0.830} \FloatTok{0.894} \FloatTok{0.873} \FloatTok{0.895}  \FloatTok{0.863}
  \CommentTok{# ... with 138 more rows, and 140 more variables: `1047` <dbl>,}
  \CommentTok{#   `1035` <dbl>, `988` <dbl>, `939` <dbl>, `936` <dbl>, `748` <dbl>,}
  \CommentTok{#   `1031` <dbl>, `903` <dbl>, `864` <dbl>, `874` <dbl>, `898` <dbl>,}
  \CommentTok{#   `1013` <dbl>, `971` <dbl>, `966` <dbl>, `866` <dbl>, `924` <dbl>,}
  \CommentTok{#   `931` <dbl>, `1007` <dbl>, `954` <dbl>, `958` <dbl>, `1037` <dbl>,}
  \CommentTok{#   `965` <dbl>, `1008` <dbl>, `1005` <dbl>, `962` <dbl>, `979` <dbl>,}
  \CommentTok{#   `881` <dbl>, `876` <dbl>, `764` <dbl>, `743` <dbl>, `987` <dbl>,}
  \CommentTok{#   `930` <dbl>, `1023` <dbl>, `928` <dbl>, `910` <dbl>, `897` <dbl>,}
  \CommentTok{#   `1036` <dbl>, `904` <dbl>, `769` <dbl>, `907` <dbl>, `821` <dbl>,}
  \CommentTok{#   `990` <dbl>, `747` <dbl>, `753` <dbl>, `843` <dbl>, `761` <dbl>,}
  \CommentTok{#   `819` <dbl>, `820` <dbl>, `802` <dbl>, `805` <dbl>, `870` <dbl>,}
  \CommentTok{#   `817` <dbl>, `1040` <dbl>, `815` <dbl>, `952` <dbl>, `974` <dbl>,}
  \CommentTok{#   `951` <dbl>, `929` <dbl>, `980` <dbl>, `911` <dbl>, `927` <dbl>,}
  \CommentTok{#   `914` <dbl>, `841` <dbl>, `912` <dbl>, `969` <dbl>, `754` <dbl>,}
  \CommentTok{#   `1053` <dbl>, `884` <dbl>, `878` <dbl>, `909` <dbl>, `810` <dbl>,}
  \CommentTok{#   `863` <dbl>, `925` <dbl>, `853` <dbl>, `857` <dbl>, `850` <dbl>,}
  \CommentTok{#   `950` <dbl>, `1027` <dbl>, `948` <dbl>, `970` <dbl>, `831` <dbl>,}
  \CommentTok{#   `813` <dbl>, `1051` <dbl>, `913` <dbl>, `1015` <dbl>, `1054` <dbl>,}
  \CommentTok{#   `937` <dbl>, `1006` <dbl>, `940` <dbl>, `827` <dbl>, `791` <dbl>,}
  \CommentTok{#   `991` <dbl>, `839` <dbl>, `818` <dbl>, `828` <dbl>, `774` <dbl>,}
  \CommentTok{#   `845` <dbl>, `797` <dbl>, `998` <dbl>, `767` <dbl>, ...}
\end{Highlighting}
\end{Shaded}

This file also contains different variables that are available in this \texttt{tibble}. The

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{progress_vars}
  \CommentTok{# A tibble: 150 x 4}
\NormalTok{     id    birthweight   EGA    acc}
     \OperatorTok{<}\NormalTok{chr}\OperatorTok{>}\StringTok{       }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{  }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}
\StringTok{   }\DecValTok{1} \DecValTok{784}          \FloatTok{2.62}  \DecValTok{38}    \FloatTok{0.792}
   \DecValTok{2} \DecValTok{1052}         \FloatTok{2.59}  \FloatTok{38.3} \FloatTok{-1.05} 
   \DecValTok{3} \DecValTok{1048}         \FloatTok{3.20}  \DecValTok{38}    \FloatTok{2.29} 
   \DecValTok{4} \DecValTok{1017}         \FloatTok{3.28}  \FloatTok{38.6}  \FloatTok{0.643}
   \DecValTok{5} \DecValTok{956}          \FloatTok{2.79}  \FloatTok{37.1}  \FloatTok{1.75} 
   \DecValTok{6} \DecValTok{1038}         \FloatTok{2.89}  \FloatTok{38.1}  \FloatTok{1.09} 
   \DecValTok{7} \DecValTok{989}          \FloatTok{2.47}  \DecValTok{38}   \FloatTok{-0.774}
   \DecValTok{8} \DecValTok{946}          \FloatTok{2.42}  \FloatTok{37.7} \FloatTok{-2.36} 
   \DecValTok{9} \DecValTok{941}          \FloatTok{2.96}  \FloatTok{36.7} \FloatTok{-3.18} 
  \DecValTok{10} \DecValTok{1024}         \FloatTok{2.61}  \FloatTok{38.6} \FloatTok{-1.12} 
  \CommentTok{# ... with 140 more rows}
\end{Highlighting}
\end{Shaded}

Clinical Variables including clinical assesment of gestational age (EGA) are available at this \texttt{tibble}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{progress_vars}
  \CommentTok{# A tibble: 150 x 4}
\NormalTok{     id    birthweight   EGA    acc}
     \OperatorTok{<}\NormalTok{chr}\OperatorTok{>}\StringTok{       }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{  }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}
\StringTok{   }\DecValTok{1} \DecValTok{784}          \FloatTok{2.62}  \DecValTok{38}    \FloatTok{0.792}
   \DecValTok{2} \DecValTok{1052}         \FloatTok{2.59}  \FloatTok{38.3} \FloatTok{-1.05} 
   \DecValTok{3} \DecValTok{1048}         \FloatTok{3.20}  \DecValTok{38}    \FloatTok{2.29} 
   \DecValTok{4} \DecValTok{1017}         \FloatTok{3.28}  \FloatTok{38.6}  \FloatTok{0.643}
   \DecValTok{5} \DecValTok{956}          \FloatTok{2.79}  \FloatTok{37.1}  \FloatTok{1.75} 
   \DecValTok{6} \DecValTok{1038}         \FloatTok{2.89}  \FloatTok{38.1}  \FloatTok{1.09} 
   \DecValTok{7} \DecValTok{989}          \FloatTok{2.47}  \DecValTok{38}   \FloatTok{-0.774}
   \DecValTok{8} \DecValTok{946}          \FloatTok{2.42}  \FloatTok{37.7} \FloatTok{-2.36} 
   \DecValTok{9} \DecValTok{941}          \FloatTok{2.96}  \FloatTok{36.7} \FloatTok{-3.18} 
  \DecValTok{10} \DecValTok{1024}         \FloatTok{2.61}  \FloatTok{38.6} \FloatTok{-1.12} 
  \CommentTok{# ... with 140 more rows}
\end{Highlighting}
\end{Shaded}

The Gestational Age (in months) is simply computed by

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{ga.progress <-}\StringTok{ }\KeywordTok{DNAmGA}\NormalTok{(progress_data)}
\NormalTok{ga.progress}
  \CommentTok{# A tibble: 150 x 5}
\NormalTok{     id    Knight Bohlin Mayne Lee  }
     \OperatorTok{<}\NormalTok{fct}\OperatorTok{>}\StringTok{  }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{lgl}\OperatorTok{>}\StringTok{  }\ErrorTok{<}\NormalTok{lgl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{lgl}\OperatorTok{>}
\StringTok{   }\DecValTok{1} \DecValTok{784}     \FloatTok{38.8} \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}   
   \DecValTok{2} \DecValTok{1052}    \FloatTok{37.2} \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}   
   \DecValTok{3} \DecValTok{1048}    \FloatTok{40.3} \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}   
   \DecValTok{4} \DecValTok{1017}    \FloatTok{39.2} \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}   
   \DecValTok{5} \DecValTok{956}     \FloatTok{38.9} \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}   
   \DecValTok{6} \DecValTok{1038}    \FloatTok{39.2} \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}   
   \DecValTok{7} \DecValTok{989}     \FloatTok{37.2} \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}   
   \DecValTok{8} \DecValTok{946}     \FloatTok{35.4} \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}   
   \DecValTok{9} \DecValTok{941}     \FloatTok{33.5} \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}   
  \DecValTok{10} \DecValTok{1024}    \FloatTok{37.4} \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}   
  \CommentTok{# ... with 140 more rows}
\end{Highlighting}
\end{Shaded}

We can compare these results with the clinical GA available in the variable EGA

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{plotDNAmAge}\NormalTok{(ga.progress}\OperatorTok{$}\NormalTok{Knight, progress_vars}\OperatorTok{$}\NormalTok{EGA, }
            \DataTypeTok{tit=}\StringTok{"GA Knight's method"}\NormalTok{, }
            \DataTypeTok{clock=}\StringTok{"GA"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{adjustwidth}{\fltoffset}{0mm}
\includegraphics{methylclock_files/figure-latex/plot_progress-1} \end{adjustwidth}

Figure 3b (only for PROGRESS dataset) in Knight et al. (2016) representing the correlation between GA acceleration and birthweight can be reproduced by

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(ggplot2)}
\NormalTok{progress_vars}\OperatorTok{$}\NormalTok{acc <-}\StringTok{ }\NormalTok{ga.progress}\OperatorTok{$}\NormalTok{Knight }\OperatorTok{-}\StringTok{ }\NormalTok{progress_vars}\OperatorTok{$}\NormalTok{EGA}
\NormalTok{p <-}\StringTok{ }\KeywordTok{ggplot}\NormalTok{(}\DataTypeTok{data=}\NormalTok{progress_vars, }\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x =}\NormalTok{ acc, }\DataTypeTok{y =}\NormalTok{ birthweight)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_point}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_smooth}\NormalTok{(}\DataTypeTok{method =} \StringTok{"lm"}\NormalTok{, }\DataTypeTok{se=}\OtherTok{FALSE}\NormalTok{, }\DataTypeTok{color=}\StringTok{"black"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{xlab}\NormalTok{(}\StringTok{"GA acceleration"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{ylab}\NormalTok{(}\StringTok{"Birthweight (kgs.)"}\NormalTok{) }
\NormalTok{p}
\end{Highlighting}
\end{Shaded}

\begin{adjustwidth}{\fltoffset}{0mm}
\includegraphics{methylclock_files/figure-latex/plotAcc-1} \end{adjustwidth}

Finally, we can also estimate the ``accelerated gestational age'' using two of the the three different estimates previously described (\texttt{accAge}, \texttt{accAge2}) by provinding information of gestational age through \texttt{age} argument. Notice that in that case \texttt{accAge3} cannot be estimates since we do not have all the CpGs required by the default reference panel to estimate cell counts for gestational age which is ``andrews and bakulski cord blood''.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{accga.progress <-}\StringTok{ }\KeywordTok{DNAmGA}\NormalTok{(progress_data, }
                         \DataTypeTok{age =}\NormalTok{ progress_vars}\OperatorTok{$}\NormalTok{EGA, }
                         \DataTypeTok{cell.count=}\OtherTok{FALSE}\NormalTok{)}
\NormalTok{accga.progress}
  \CommentTok{# A tibble: 150 x 8}
\NormalTok{     id    Knight ageAcc.Knight ageAcc2.Knight Bohlin Mayne Lee     age}
     \OperatorTok{<}\NormalTok{fct}\OperatorTok{>}\StringTok{  }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{         }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{          }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{lgl}\OperatorTok{>}\StringTok{  }\ErrorTok{<}\NormalTok{lgl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{lgl}\OperatorTok{>}\StringTok{ }\ErrorTok{<}\NormalTok{dbl}\OperatorTok{>}
\StringTok{   }\DecValTok{1} \DecValTok{784}     \FloatTok{38.8}         \FloatTok{0.792}          \FloatTok{1.27}  \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}     \DecValTok{38}  
   \DecValTok{2} \DecValTok{1052}    \FloatTok{37.2}        \FloatTok{-1.05}          \FloatTok{-0.488} \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}     \FloatTok{38.3}
   \DecValTok{3} \DecValTok{1048}    \FloatTok{40.3}         \FloatTok{2.29}           \FloatTok{2.77}  \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}     \DecValTok{38}  
   \DecValTok{4} \DecValTok{1017}    \FloatTok{39.2}         \FloatTok{0.643}          \FloatTok{1.28}  \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}     \FloatTok{38.6}
   \DecValTok{5} \DecValTok{956}     \FloatTok{38.9}         \FloatTok{1.75}           \FloatTok{1.99}  \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}     \FloatTok{37.1}
   \DecValTok{6} \DecValTok{1038}    \FloatTok{39.2}         \FloatTok{1.09}           \FloatTok{1.61}  \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}     \FloatTok{38.1}
   \DecValTok{7} \DecValTok{989}     \FloatTok{37.2}        \FloatTok{-0.774}         \FloatTok{-0.292} \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}     \DecValTok{38}  
   \DecValTok{8} \DecValTok{946}     \FloatTok{35.4}        \FloatTok{-2.36}          \FloatTok{-1.96}  \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}     \FloatTok{37.7}
   \DecValTok{9} \DecValTok{941}     \FloatTok{33.5}        \FloatTok{-3.18}          \FloatTok{-3.06}  \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}     \FloatTok{36.7}
  \DecValTok{10} \DecValTok{1024}    \FloatTok{37.4}        \FloatTok{-1.12}          \FloatTok{-0.486} \OtherTok{NA}     \OtherTok{NA}    \OtherTok{NA}     \FloatTok{38.6}
  \CommentTok{# ... with 140 more rows}
\end{Highlighting}
\end{Shaded}

\hypertarget{correlation-among-dnam-clocks}{%
\section{Correlation among DNAm clocks}\label{correlation-among-dnam-clocks}}

We can compute the correlation among biological clocks using the function \texttt{plotCorClocks} that requires the package \texttt{ggplot2} and \texttt{ggpubr} to be installed in your computer.

We can obtain, for instance, the correlation among the clocks estimated for the healthy individuals study previosuly analyze (GEO accession number GSE58045) by simply executing:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{plotCorClocks}\NormalTok{(age.gse58045)}
\end{Highlighting}
\end{Shaded}

\begin{adjustwidth}{\fltoffset}{0mm}
\includegraphics{methylclock_files/figure-latex/plotCorClockHealth-1} \end{adjustwidth}

\hypertarget{references}{%
\section*{References}\label{references}}
\addcontentsline{toc}{section}{References}

\hypertarget{refs}{}
\leavevmode\hypertarget{ref-alfonso2018}{}%
Alfonso, Gerardo, and Juan R Gonzalez. 2018. ``Bayesian Neural Networks Improve Methylation Age Estimates.'' \emph{bioRxiv} XX (X): XX.

\leavevmode\hypertarget{ref-bakulski2016dna}{}%
Bakulski, Kelly M, Jason I Feinberg, Shan V Andrews, Jack Yang, Shannon Brown, Stephanie L. McKenney, Frank Witter, Jeremy Walston, Andrew P Feinberg, and M Daniele Fallin. 2016. ``DNA Methylation of Cord Blood Cell Types: Applications for Mixed Cell Birth Studies.'' \emph{Epigenetics} 11 (5): 354--62.

\leavevmode\hypertarget{ref-bohlin2016prediction}{}%
Bohlin, Jon, Siri Eldevik HÃ¥berg, Per Magnus, Sarah E Reese, HÃ¥kon K Gjessing, Maria Christine Magnus, Christine Louise Parr, CM Page, Stephanie J London, and Wenche Nystad. 2016. ``Prediction of Gestational Age Based on Genome-Wide Differentially Methylated Regions.'' \emph{Genome Biology} 17 (1): 207.

\leavevmode\hypertarget{ref-chen2017epigenome}{}%
Chen, Wei, Ting Wang, Maria Pino-Yanes, Erick Forno, Liming Liang, Qi Yan, Donglei Hu, et al. 2017. ``An Epigenome-Wide Association Study of Total Serum Ige in Hispanic Children.'' \emph{Journal of Allergy and Clinical Immunology} 140 (2): 571--77.

\leavevmode\hypertarget{ref-de2015nucleated}{}%
Goede, Olivia M de, Hamid R Razzaghian, E Magda Price, Meaghan J Jones, Michael S Kobor, Wendy P Robinson, and Pascal M Lavoie. 2015. ``Nucleated Red Blood Cells Impact Dna Methylation and Expression Analyses of Cord Blood Hematopoietic Cells.'' \emph{Clinical Epigenetics} 7 (1): 95.

\leavevmode\hypertarget{ref-hannum2013genome}{}%
Hannum, Gregory, Justin Guinney, Ling Zhao, Li Zhang, Guy Hughes, SriniVas Sadda, Brandy Klotzle, et al. 2013. ``Genome-Wide Methylation Profiles Reveal Quantitative Views of Human Aging Rates.'' \emph{Molecular Cell} 49 (2): 359--67.

\leavevmode\hypertarget{ref-horvath2013dna}{}%
Horvath, Steve. 2013. ``DNA Methylation Age of Human Tissues and Cell Types.'' \emph{Genome Biology} 14 (10): 3156.

\leavevmode\hypertarget{ref-horvath2018epigenetic}{}%
Horvath, Steve, Junko Oshima, George M Martin, Ake T Lu, Austin Quach, Howard Cohen, Sarah Felton, et al. 2018. ``Epigenetic Clock for Skin and Blood Cells Applied to Hutchinson Gilford Progeria Syndrome and Ex Vivo Studies.'' \emph{Aging (Albany NY)} 10 (7): 1758.

\leavevmode\hypertarget{ref-knight2016epigenetic}{}%
Knight, Anna K, Jeffrey M Craig, Christiane Theda, Marie BÃ¦kvad-Hansen, Jonas Bybjerg-Grauholm, Christine S Hansen, Mads V Hollegaard, et al. 2016. ``An Epigenetic Clock for Gestational Age at Birth Based on Blood Methylation Data.'' \emph{Genome Biology} 17 (1): 206.

\leavevmode\hypertarget{ref-lee2019placental}{}%
Lee, Yunsung, Sanaa Choufani, Rosanna Weksberg, Samantha L Wilson, Victor Yuan, Amber Burt, Carmen Marsit, et al. 2019. ``Placental Epigenetic Clocks: Estimating Gestational Age Using Placental Dna Methylation Levels.'' \emph{Aging (Albany NY)} 11 (12): 4238.

\leavevmode\hypertarget{ref-levine2018epigenetic}{}%
Levine, Morgan E, Ake T Lu, Austin Quach, Brian H Chen, Themistocles L Assimes, Stefania Bandinelli, Lifang Hou, et al. 2018. ``An Epigenetic Biomarker of Aging for Lifespan and Healthspan.'' \emph{Aging (Albany NY)} 10 (4): 573.

\leavevmode\hypertarget{ref-mayne2017accelerated}{}%
Mayne, Benjamin T, Shalem Y Leemaqz, Alicia K Smith, James Breen, Claire T Roberts, and Tina Bianco-Miotto. 2017. ``Accelerated Placental Aging in Early Onset Preeclampsia Pregnancies Identified by Dna Methylation.'' \emph{Epigenomics} 9 (3): 279--89.

\leavevmode\hypertarget{ref-mcewen2019pedbe}{}%
McEwen, Lisa M, Kieran J O?Donnell, Megan G McGill, Rachel D Edgar, Meaghan J Jones, Julia L MacIsaac, David Tse Shen Lin, et al. 2019. ``The Pedbe Clock Accurately Estimates Dna Methylation Age in Pediatric Buccal Cells.'' \emph{Proceedings of the National Academy of Sciences}, 201820843.

\leavevmode\hypertarget{ref-min2018meffil}{}%
Min, JL, G Hemani, G Davey Smith, C Relton, M Suderman, and John Hancock. 2018. ``Meffil: Efficient Normalization and Analysis of Very Large Dna Methylation Datasets.'' \emph{Bioinformatics}.

\leavevmode\hypertarget{ref-reinius2012differential}{}%
Reinius, Lovisa E, Nathalie Acevedo, Maaike Joerink, GÃ¶ran Pershagen, Sven-Erik DahlÃ©n, Dario Greco, Cilla SÃ¶derhÃ¤ll, Annika Scheynius, and Juha Kere. 2012. ``Differential Dna Methylation in Purified Human Blood Cells: Implications for Cell Lineage and Studies on Disease Susceptibility.'' \emph{PloS One} 7 (7): e41361.

\leavevmode\hypertarget{ref-slieker2013identification}{}%
Slieker, Roderick C, Steffan D Bos, Jelle J Goeman, Judith VMG BovÃ©e, Rudolf P Talens, Ruud van der Breggen, H Eka D Suchiman, et al. 2013. ``Identification and Systematic Annotation of Tissue-Specific Differentially Methylated Regions Using the Illumina 450k Array.'' \emph{Epigenetics \& Chromatin} 6 (1): 26.

\leavevmode\hypertarget{ref-teschendorff2012beta}{}%
Teschendorff, Andrew E, Francesco Marabita, Matthias Lechner, Thomas Bartlett, Jesper Tegner, David Gomez-Cabrero, and Stephan Beck. 2012. ``A Beta-Mixture Quantile Normalization Method for Correcting Probe Design Bias in Illumina Infinium 450 K Dna Methylation Data.'' \emph{Bioinformatics} 29 (2): 189--96.

\leavevmode\hypertarget{ref-wang2015systematic}{}%
Wang, Ting, Weihua Guan, Jerome Lin, Nadia Boutaoui, Glorisa Canino, Jianhua Luo, Juan Carlos CeledÃ³n, and Wei Chen. 2015. ``A Systematic Study of Normalization Methods for Infinium 450K Methylation Data Using Whole-Genome Bisulfite Sequencing Data.'' \emph{Epigenetics} 10 (7): 662--69.

\end{document}
